// @prisma/client/enums.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// @ts-nocheck 
/*
* This file exports all enum related types from the schema.
*
* 🟢 You can import this file directly.
*/

export const UserRole = {
  GUEST: 'GUEST',
  CONSUMER: 'CONSUMER',
  CREATOR: 'CREATOR',
  ADMIN: 'ADMIN'
} as const

export type UserRole = (typeof UserRole)[keyof typeof UserRole]


export const SessionKind = {
  GUEST: 'GUEST',
  AUTHENTICATED: 'AUTHENTICATED'
} as const

export type SessionKind = (typeof SessionKind)[keyof typeof SessionKind]


export const JourneyStatus = {
  NOT_STARTED: 'NOT_STARTED',
  IN_PROGRESS: 'IN_PROGRESS',
  COMPLETED: 'COMPLETED',
  ABANDONED: 'ABANDONED'
} as const

export type JourneyStatus = (typeof JourneyStatus)[keyof typeof JourneyStatus]


export const StoryStatus = {
  DRAFT: 'DRAFT',
  PENDING: 'PENDING',
  APPROVED: 'APPROVED',
  REJECTED: 'REJECTED',
  ARCHIVED: 'ARCHIVED'
} as const

export type StoryStatus = (typeof StoryStatus)[keyof typeof StoryStatus]


export const StoryVisibility = {
  PRIVATE: 'PRIVATE',
  UNLISTED: 'UNLISTED',
  PUBLIC: 'PUBLIC'
} as const

export type StoryVisibility = (typeof StoryVisibility)[keyof typeof StoryVisibility]


export const StoryNodeType = {
  NARRATIVE: 'NARRATIVE',
  DECISION: 'DECISION',
  RESOLUTION: 'RESOLUTION'
} as const

export type StoryNodeType = (typeof StoryNodeType)[keyof typeof StoryNodeType]


export const CreatorProfileStatus = {
  DRAFT: 'DRAFT',
  REVIEW: 'REVIEW',
  ACTIVE: 'ACTIVE',
  SUSPENDED: 'SUSPENDED'
} as const

export type CreatorProfileStatus = (typeof CreatorProfileStatus)[keyof typeof CreatorProfileStatus]


export const StoryOwnershipStatus = {
  CREATOR_DRAFT: 'CREATOR_DRAFT',
  PENDING_TRANSFER: 'PENDING_TRANSFER',
  PLATFORM_OWNED: 'PLATFORM_OWNED',
  RETURNED: 'RETURNED'
} as const

export type StoryOwnershipStatus = (typeof StoryOwnershipStatus)[keyof typeof StoryOwnershipStatus]
