// @prisma/client/models/StoryTransition.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// @ts-nocheck 
/*
 * This file exports the `StoryTransition` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/library"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model StoryTransition
 * 
 */
export type StoryTransitionModel = runtime.Types.Result.DefaultSelection<Prisma.$StoryTransitionPayload>

export type AggregateStoryTransition = {
  _count: StoryTransitionCountAggregateOutputType | null
  _avg: StoryTransitionAvgAggregateOutputType | null
  _sum: StoryTransitionSumAggregateOutputType | null
  _min: StoryTransitionMinAggregateOutputType | null
  _max: StoryTransitionMaxAggregateOutputType | null
}

export type StoryTransitionAvgAggregateOutputType = {
  ordering: number | null
}

export type StoryTransitionSumAggregateOutputType = {
  ordering: number | null
}

export type StoryTransitionMinAggregateOutputType = {
  id: string | null
  storyId: string | null
  fromNodeId: string | null
  pathId: string | null
  toNodeId: string | null
  ordering: number | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryTransitionMaxAggregateOutputType = {
  id: string | null
  storyId: string | null
  fromNodeId: string | null
  pathId: string | null
  toNodeId: string | null
  ordering: number | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryTransitionCountAggregateOutputType = {
  id: number
  storyId: number
  fromNodeId: number
  pathId: number
  toNodeId: number
  ordering: number
  condition: number
  effect: number
  createdAt: number
  updatedAt: number
  _all: number
}


export type StoryTransitionAvgAggregateInputType = {
  ordering?: true
}

export type StoryTransitionSumAggregateInputType = {
  ordering?: true
}

export type StoryTransitionMinAggregateInputType = {
  id?: true
  storyId?: true
  fromNodeId?: true
  pathId?: true
  toNodeId?: true
  ordering?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryTransitionMaxAggregateInputType = {
  id?: true
  storyId?: true
  fromNodeId?: true
  pathId?: true
  toNodeId?: true
  ordering?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryTransitionCountAggregateInputType = {
  id?: true
  storyId?: true
  fromNodeId?: true
  pathId?: true
  toNodeId?: true
  ordering?: true
  condition?: true
  effect?: true
  createdAt?: true
  updatedAt?: true
  _all?: true
}

export type StoryTransitionAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryTransition to aggregate.
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryTransitions to fetch.
   */
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryTransitions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryTransitions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned StoryTransitions
  **/
  _count?: true | StoryTransitionCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: StoryTransitionAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: StoryTransitionSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: StoryTransitionMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: StoryTransitionMaxAggregateInputType
}

export type GetStoryTransitionAggregateType<T extends StoryTransitionAggregateArgs> = {
      [P in keyof T & keyof AggregateStoryTransition]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateStoryTransition[P]>
    : Prisma.GetScalarType<T[P], AggregateStoryTransition[P]>
}




export type StoryTransitionGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryTransitionWhereInput
  orderBy?: Prisma.StoryTransitionOrderByWithAggregationInput | Prisma.StoryTransitionOrderByWithAggregationInput[]
  by: Prisma.StoryTransitionScalarFieldEnum[] | Prisma.StoryTransitionScalarFieldEnum
  having?: Prisma.StoryTransitionScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: StoryTransitionCountAggregateInputType | true
  _avg?: StoryTransitionAvgAggregateInputType
  _sum?: StoryTransitionSumAggregateInputType
  _min?: StoryTransitionMinAggregateInputType
  _max?: StoryTransitionMaxAggregateInputType
}

export type StoryTransitionGroupByOutputType = {
  id: string
  storyId: string
  fromNodeId: string
  pathId: string
  toNodeId: string | null
  ordering: number | null
  condition: runtime.JsonValue | null
  effect: runtime.JsonValue | null
  createdAt: Date
  updatedAt: Date
  _count: StoryTransitionCountAggregateOutputType | null
  _avg: StoryTransitionAvgAggregateOutputType | null
  _sum: StoryTransitionSumAggregateOutputType | null
  _min: StoryTransitionMinAggregateOutputType | null
  _max: StoryTransitionMaxAggregateOutputType | null
}

type GetStoryTransitionGroupByPayload<T extends StoryTransitionGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<StoryTransitionGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof StoryTransitionGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], StoryTransitionGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], StoryTransitionGroupByOutputType[P]>
      }
    >
  >



export type StoryTransitionWhereInput = {
  AND?: Prisma.StoryTransitionWhereInput | Prisma.StoryTransitionWhereInput[]
  OR?: Prisma.StoryTransitionWhereInput[]
  NOT?: Prisma.StoryTransitionWhereInput | Prisma.StoryTransitionWhereInput[]
  id?: Prisma.StringFilter<"StoryTransition"> | string
  storyId?: Prisma.StringFilter<"StoryTransition"> | string
  fromNodeId?: Prisma.StringFilter<"StoryTransition"> | string
  pathId?: Prisma.StringFilter<"StoryTransition"> | string
  toNodeId?: Prisma.StringNullableFilter<"StoryTransition"> | string | null
  ordering?: Prisma.IntNullableFilter<"StoryTransition"> | number | null
  condition?: Prisma.JsonNullableFilter<"StoryTransition">
  effect?: Prisma.JsonNullableFilter<"StoryTransition">
  createdAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  fromNode?: Prisma.XOR<Prisma.StoryNodeScalarRelationFilter, Prisma.StoryNodeWhereInput>
  toNode?: Prisma.XOR<Prisma.StoryNodeNullableScalarRelationFilter, Prisma.StoryNodeWhereInput> | null
  path?: Prisma.XOR<Prisma.StoryPathScalarRelationFilter, Prisma.StoryPathWhereInput>
}

export type StoryTransitionOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  fromNodeId?: Prisma.SortOrder
  pathId?: Prisma.SortOrder
  toNodeId?: Prisma.SortOrderInput | Prisma.SortOrder
  ordering?: Prisma.SortOrderInput | Prisma.SortOrder
  condition?: Prisma.SortOrderInput | Prisma.SortOrder
  effect?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  story?: Prisma.TwineStoryOrderByWithRelationInput
  fromNode?: Prisma.StoryNodeOrderByWithRelationInput
  toNode?: Prisma.StoryNodeOrderByWithRelationInput
  path?: Prisma.StoryPathOrderByWithRelationInput
}

export type StoryTransitionWhereUniqueInput = Prisma.AtLeast<{
  id?: string
  storyId_fromNodeId_pathId_toNodeId?: Prisma.StoryTransitionStoryIdFromNodeIdPathIdToNodeIdCompoundUniqueInput
  AND?: Prisma.StoryTransitionWhereInput | Prisma.StoryTransitionWhereInput[]
  OR?: Prisma.StoryTransitionWhereInput[]
  NOT?: Prisma.StoryTransitionWhereInput | Prisma.StoryTransitionWhereInput[]
  storyId?: Prisma.StringFilter<"StoryTransition"> | string
  fromNodeId?: Prisma.StringFilter<"StoryTransition"> | string
  pathId?: Prisma.StringFilter<"StoryTransition"> | string
  toNodeId?: Prisma.StringNullableFilter<"StoryTransition"> | string | null
  ordering?: Prisma.IntNullableFilter<"StoryTransition"> | number | null
  condition?: Prisma.JsonNullableFilter<"StoryTransition">
  effect?: Prisma.JsonNullableFilter<"StoryTransition">
  createdAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  fromNode?: Prisma.XOR<Prisma.StoryNodeScalarRelationFilter, Prisma.StoryNodeWhereInput>
  toNode?: Prisma.XOR<Prisma.StoryNodeNullableScalarRelationFilter, Prisma.StoryNodeWhereInput> | null
  path?: Prisma.XOR<Prisma.StoryPathScalarRelationFilter, Prisma.StoryPathWhereInput>
}, "id" | "storyId_fromNodeId_pathId_toNodeId">

export type StoryTransitionOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  fromNodeId?: Prisma.SortOrder
  pathId?: Prisma.SortOrder
  toNodeId?: Prisma.SortOrderInput | Prisma.SortOrder
  ordering?: Prisma.SortOrderInput | Prisma.SortOrder
  condition?: Prisma.SortOrderInput | Prisma.SortOrder
  effect?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  _count?: Prisma.StoryTransitionCountOrderByAggregateInput
  _avg?: Prisma.StoryTransitionAvgOrderByAggregateInput
  _max?: Prisma.StoryTransitionMaxOrderByAggregateInput
  _min?: Prisma.StoryTransitionMinOrderByAggregateInput
  _sum?: Prisma.StoryTransitionSumOrderByAggregateInput
}

export type StoryTransitionScalarWhereWithAggregatesInput = {
  AND?: Prisma.StoryTransitionScalarWhereWithAggregatesInput | Prisma.StoryTransitionScalarWhereWithAggregatesInput[]
  OR?: Prisma.StoryTransitionScalarWhereWithAggregatesInput[]
  NOT?: Prisma.StoryTransitionScalarWhereWithAggregatesInput | Prisma.StoryTransitionScalarWhereWithAggregatesInput[]
  id?: Prisma.StringWithAggregatesFilter<"StoryTransition"> | string
  storyId?: Prisma.StringWithAggregatesFilter<"StoryTransition"> | string
  fromNodeId?: Prisma.StringWithAggregatesFilter<"StoryTransition"> | string
  pathId?: Prisma.StringWithAggregatesFilter<"StoryTransition"> | string
  toNodeId?: Prisma.StringNullableWithAggregatesFilter<"StoryTransition"> | string | null
  ordering?: Prisma.IntNullableWithAggregatesFilter<"StoryTransition"> | number | null
  condition?: Prisma.JsonNullableWithAggregatesFilter<"StoryTransition">
  effect?: Prisma.JsonNullableWithAggregatesFilter<"StoryTransition">
  createdAt?: Prisma.DateTimeWithAggregatesFilter<"StoryTransition"> | Date | string
  updatedAt?: Prisma.DateTimeWithAggregatesFilter<"StoryTransition"> | Date | string
}

export type StoryTransitionCreateInput = {
  id?: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutTransitionsInput
  fromNode: Prisma.StoryNodeCreateNestedOneWithoutOutgoingInput
  toNode?: Prisma.StoryNodeCreateNestedOneWithoutIncomingInput
  path: Prisma.StoryPathCreateNestedOneWithoutTransitionsInput
}

export type StoryTransitionUncheckedCreateInput = {
  id?: string
  storyId: string
  fromNodeId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutTransitionsNestedInput
  fromNode?: Prisma.StoryNodeUpdateOneRequiredWithoutOutgoingNestedInput
  toNode?: Prisma.StoryNodeUpdateOneWithoutIncomingNestedInput
  path?: Prisma.StoryPathUpdateOneRequiredWithoutTransitionsNestedInput
}

export type StoryTransitionUncheckedUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionCreateManyInput = {
  id?: string
  storyId: string
  fromNodeId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionUpdateManyMutationInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUncheckedUpdateManyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionListRelationFilter = {
  every?: Prisma.StoryTransitionWhereInput
  some?: Prisma.StoryTransitionWhereInput
  none?: Prisma.StoryTransitionWhereInput
}

export type StoryTransitionOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type StoryTransitionStoryIdFromNodeIdPathIdToNodeIdCompoundUniqueInput = {
  storyId: string
  fromNodeId: string
  pathId: string
  toNodeId: string
}

export type StoryTransitionCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  fromNodeId?: Prisma.SortOrder
  pathId?: Prisma.SortOrder
  toNodeId?: Prisma.SortOrder
  ordering?: Prisma.SortOrder
  condition?: Prisma.SortOrder
  effect?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryTransitionAvgOrderByAggregateInput = {
  ordering?: Prisma.SortOrder
}

export type StoryTransitionMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  fromNodeId?: Prisma.SortOrder
  pathId?: Prisma.SortOrder
  toNodeId?: Prisma.SortOrder
  ordering?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryTransitionMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  fromNodeId?: Prisma.SortOrder
  pathId?: Prisma.SortOrder
  toNodeId?: Prisma.SortOrder
  ordering?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryTransitionSumOrderByAggregateInput = {
  ordering?: Prisma.SortOrder
}

export type StoryTransitionCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput> | Prisma.StoryTransitionCreateWithoutStoryInput[] | Prisma.StoryTransitionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutStoryInput | Prisma.StoryTransitionCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryTransitionCreateManyStoryInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUncheckedCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput> | Prisma.StoryTransitionCreateWithoutStoryInput[] | Prisma.StoryTransitionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutStoryInput | Prisma.StoryTransitionCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryTransitionCreateManyStoryInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput> | Prisma.StoryTransitionCreateWithoutStoryInput[] | Prisma.StoryTransitionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutStoryInput | Prisma.StoryTransitionCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryTransitionCreateManyStoryInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutStoryInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionUncheckedUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput> | Prisma.StoryTransitionCreateWithoutStoryInput[] | Prisma.StoryTransitionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutStoryInput | Prisma.StoryTransitionCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryTransitionCreateManyStoryInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutStoryInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionCreateNestedManyWithoutFromNodeInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput> | Prisma.StoryTransitionCreateWithoutFromNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyFromNodeInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionCreateNestedManyWithoutToNodeInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput> | Prisma.StoryTransitionCreateWithoutToNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyToNodeInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUncheckedCreateNestedManyWithoutFromNodeInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput> | Prisma.StoryTransitionCreateWithoutFromNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyFromNodeInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUncheckedCreateNestedManyWithoutToNodeInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput> | Prisma.StoryTransitionCreateWithoutToNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyToNodeInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUpdateManyWithoutFromNodeNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput> | Prisma.StoryTransitionCreateWithoutFromNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutFromNodeInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutFromNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyFromNodeInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutFromNodeInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutFromNodeInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutFromNodeInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutFromNodeInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionUpdateManyWithoutToNodeNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput> | Prisma.StoryTransitionCreateWithoutToNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutToNodeInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutToNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyToNodeInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutToNodeInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutToNodeInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutToNodeInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutToNodeInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionUncheckedUpdateManyWithoutFromNodeNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput> | Prisma.StoryTransitionCreateWithoutFromNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutFromNodeInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutFromNodeInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutFromNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyFromNodeInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutFromNodeInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutFromNodeInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutFromNodeInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutFromNodeInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionUncheckedUpdateManyWithoutToNodeNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput> | Prisma.StoryTransitionCreateWithoutToNodeInput[] | Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput | Prisma.StoryTransitionCreateOrConnectWithoutToNodeInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutToNodeInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutToNodeInput[]
  createMany?: Prisma.StoryTransitionCreateManyToNodeInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutToNodeInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutToNodeInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutToNodeInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutToNodeInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionCreateNestedManyWithoutPathInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput> | Prisma.StoryTransitionCreateWithoutPathInput[] | Prisma.StoryTransitionUncheckedCreateWithoutPathInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutPathInput | Prisma.StoryTransitionCreateOrConnectWithoutPathInput[]
  createMany?: Prisma.StoryTransitionCreateManyPathInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUncheckedCreateNestedManyWithoutPathInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput> | Prisma.StoryTransitionCreateWithoutPathInput[] | Prisma.StoryTransitionUncheckedCreateWithoutPathInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutPathInput | Prisma.StoryTransitionCreateOrConnectWithoutPathInput[]
  createMany?: Prisma.StoryTransitionCreateManyPathInputEnvelope
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
}

export type StoryTransitionUpdateManyWithoutPathNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput> | Prisma.StoryTransitionCreateWithoutPathInput[] | Prisma.StoryTransitionUncheckedCreateWithoutPathInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutPathInput | Prisma.StoryTransitionCreateOrConnectWithoutPathInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutPathInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutPathInput[]
  createMany?: Prisma.StoryTransitionCreateManyPathInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutPathInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutPathInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutPathInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutPathInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionUncheckedUpdateManyWithoutPathNestedInput = {
  create?: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput> | Prisma.StoryTransitionCreateWithoutPathInput[] | Prisma.StoryTransitionUncheckedCreateWithoutPathInput[]
  connectOrCreate?: Prisma.StoryTransitionCreateOrConnectWithoutPathInput | Prisma.StoryTransitionCreateOrConnectWithoutPathInput[]
  upsert?: Prisma.StoryTransitionUpsertWithWhereUniqueWithoutPathInput | Prisma.StoryTransitionUpsertWithWhereUniqueWithoutPathInput[]
  createMany?: Prisma.StoryTransitionCreateManyPathInputEnvelope
  set?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  disconnect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  delete?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  connect?: Prisma.StoryTransitionWhereUniqueInput | Prisma.StoryTransitionWhereUniqueInput[]
  update?: Prisma.StoryTransitionUpdateWithWhereUniqueWithoutPathInput | Prisma.StoryTransitionUpdateWithWhereUniqueWithoutPathInput[]
  updateMany?: Prisma.StoryTransitionUpdateManyWithWhereWithoutPathInput | Prisma.StoryTransitionUpdateManyWithWhereWithoutPathInput[]
  deleteMany?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
}

export type StoryTransitionCreateWithoutStoryInput = {
  id?: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  fromNode: Prisma.StoryNodeCreateNestedOneWithoutOutgoingInput
  toNode?: Prisma.StoryNodeCreateNestedOneWithoutIncomingInput
  path: Prisma.StoryPathCreateNestedOneWithoutTransitionsInput
}

export type StoryTransitionUncheckedCreateWithoutStoryInput = {
  id?: string
  fromNodeId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionCreateOrConnectWithoutStoryInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput>
}

export type StoryTransitionCreateManyStoryInputEnvelope = {
  data: Prisma.StoryTransitionCreateManyStoryInput | Prisma.StoryTransitionCreateManyStoryInput[]
  skipDuplicates?: boolean
}

export type StoryTransitionUpsertWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutStoryInput, Prisma.StoryTransitionUncheckedUpdateWithoutStoryInput>
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutStoryInput, Prisma.StoryTransitionUncheckedCreateWithoutStoryInput>
}

export type StoryTransitionUpdateWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutStoryInput, Prisma.StoryTransitionUncheckedUpdateWithoutStoryInput>
}

export type StoryTransitionUpdateManyWithWhereWithoutStoryInput = {
  where: Prisma.StoryTransitionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyWithoutStoryInput>
}

export type StoryTransitionScalarWhereInput = {
  AND?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
  OR?: Prisma.StoryTransitionScalarWhereInput[]
  NOT?: Prisma.StoryTransitionScalarWhereInput | Prisma.StoryTransitionScalarWhereInput[]
  id?: Prisma.StringFilter<"StoryTransition"> | string
  storyId?: Prisma.StringFilter<"StoryTransition"> | string
  fromNodeId?: Prisma.StringFilter<"StoryTransition"> | string
  pathId?: Prisma.StringFilter<"StoryTransition"> | string
  toNodeId?: Prisma.StringNullableFilter<"StoryTransition"> | string | null
  ordering?: Prisma.IntNullableFilter<"StoryTransition"> | number | null
  condition?: Prisma.JsonNullableFilter<"StoryTransition">
  effect?: Prisma.JsonNullableFilter<"StoryTransition">
  createdAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryTransition"> | Date | string
}

export type StoryTransitionCreateWithoutFromNodeInput = {
  id?: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutTransitionsInput
  toNode?: Prisma.StoryNodeCreateNestedOneWithoutIncomingInput
  path: Prisma.StoryPathCreateNestedOneWithoutTransitionsInput
}

export type StoryTransitionUncheckedCreateWithoutFromNodeInput = {
  id?: string
  storyId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionCreateOrConnectWithoutFromNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput>
}

export type StoryTransitionCreateManyFromNodeInputEnvelope = {
  data: Prisma.StoryTransitionCreateManyFromNodeInput | Prisma.StoryTransitionCreateManyFromNodeInput[]
  skipDuplicates?: boolean
}

export type StoryTransitionCreateWithoutToNodeInput = {
  id?: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutTransitionsInput
  fromNode: Prisma.StoryNodeCreateNestedOneWithoutOutgoingInput
  path: Prisma.StoryPathCreateNestedOneWithoutTransitionsInput
}

export type StoryTransitionUncheckedCreateWithoutToNodeInput = {
  id?: string
  storyId: string
  fromNodeId: string
  pathId: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionCreateOrConnectWithoutToNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput>
}

export type StoryTransitionCreateManyToNodeInputEnvelope = {
  data: Prisma.StoryTransitionCreateManyToNodeInput | Prisma.StoryTransitionCreateManyToNodeInput[]
  skipDuplicates?: boolean
}

export type StoryTransitionUpsertWithWhereUniqueWithoutFromNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedUpdateWithoutFromNodeInput>
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutFromNodeInput>
}

export type StoryTransitionUpdateWithWhereUniqueWithoutFromNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutFromNodeInput, Prisma.StoryTransitionUncheckedUpdateWithoutFromNodeInput>
}

export type StoryTransitionUpdateManyWithWhereWithoutFromNodeInput = {
  where: Prisma.StoryTransitionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyWithoutFromNodeInput>
}

export type StoryTransitionUpsertWithWhereUniqueWithoutToNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutToNodeInput, Prisma.StoryTransitionUncheckedUpdateWithoutToNodeInput>
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutToNodeInput, Prisma.StoryTransitionUncheckedCreateWithoutToNodeInput>
}

export type StoryTransitionUpdateWithWhereUniqueWithoutToNodeInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutToNodeInput, Prisma.StoryTransitionUncheckedUpdateWithoutToNodeInput>
}

export type StoryTransitionUpdateManyWithWhereWithoutToNodeInput = {
  where: Prisma.StoryTransitionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyWithoutToNodeInput>
}

export type StoryTransitionCreateWithoutPathInput = {
  id?: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutTransitionsInput
  fromNode: Prisma.StoryNodeCreateNestedOneWithoutOutgoingInput
  toNode?: Prisma.StoryNodeCreateNestedOneWithoutIncomingInput
}

export type StoryTransitionUncheckedCreateWithoutPathInput = {
  id?: string
  storyId: string
  fromNodeId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionCreateOrConnectWithoutPathInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput>
}

export type StoryTransitionCreateManyPathInputEnvelope = {
  data: Prisma.StoryTransitionCreateManyPathInput | Prisma.StoryTransitionCreateManyPathInput[]
  skipDuplicates?: boolean
}

export type StoryTransitionUpsertWithWhereUniqueWithoutPathInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutPathInput, Prisma.StoryTransitionUncheckedUpdateWithoutPathInput>
  create: Prisma.XOR<Prisma.StoryTransitionCreateWithoutPathInput, Prisma.StoryTransitionUncheckedCreateWithoutPathInput>
}

export type StoryTransitionUpdateWithWhereUniqueWithoutPathInput = {
  where: Prisma.StoryTransitionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateWithoutPathInput, Prisma.StoryTransitionUncheckedUpdateWithoutPathInput>
}

export type StoryTransitionUpdateManyWithWhereWithoutPathInput = {
  where: Prisma.StoryTransitionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyWithoutPathInput>
}

export type StoryTransitionCreateManyStoryInput = {
  id?: string
  fromNodeId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  fromNode?: Prisma.StoryNodeUpdateOneRequiredWithoutOutgoingNestedInput
  toNode?: Prisma.StoryNodeUpdateOneWithoutIncomingNestedInput
  path?: Prisma.StoryPathUpdateOneRequiredWithoutTransitionsNestedInput
}

export type StoryTransitionUncheckedUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUncheckedUpdateManyWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionCreateManyFromNodeInput = {
  id?: string
  storyId: string
  pathId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionCreateManyToNodeInput = {
  id?: string
  storyId: string
  fromNodeId: string
  pathId: string
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionUpdateWithoutFromNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutTransitionsNestedInput
  toNode?: Prisma.StoryNodeUpdateOneWithoutIncomingNestedInput
  path?: Prisma.StoryPathUpdateOneRequiredWithoutTransitionsNestedInput
}

export type StoryTransitionUncheckedUpdateWithoutFromNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUncheckedUpdateManyWithoutFromNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUpdateWithoutToNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutTransitionsNestedInput
  fromNode?: Prisma.StoryNodeUpdateOneRequiredWithoutOutgoingNestedInput
  path?: Prisma.StoryPathUpdateOneRequiredWithoutTransitionsNestedInput
}

export type StoryTransitionUncheckedUpdateWithoutToNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUncheckedUpdateManyWithoutToNodeInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  pathId?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionCreateManyPathInput = {
  id?: string
  storyId: string
  fromNodeId: string
  toNodeId?: string | null
  ordering?: number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryTransitionUpdateWithoutPathInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutTransitionsNestedInput
  fromNode?: Prisma.StoryNodeUpdateOneRequiredWithoutOutgoingNestedInput
  toNode?: Prisma.StoryNodeUpdateOneWithoutIncomingNestedInput
}

export type StoryTransitionUncheckedUpdateWithoutPathInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryTransitionUncheckedUpdateManyWithoutPathInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  fromNodeId?: Prisma.StringFieldUpdateOperationsInput | string
  toNodeId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  ordering?: Prisma.NullableIntFieldUpdateOperationsInput | number | null
  condition?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  effect?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}



export type StoryTransitionSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  fromNodeId?: boolean
  pathId?: boolean
  toNodeId?: boolean
  ordering?: boolean
  condition?: boolean
  effect?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyTransition"]>

export type StoryTransitionSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  fromNodeId?: boolean
  pathId?: boolean
  toNodeId?: boolean
  ordering?: boolean
  condition?: boolean
  effect?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyTransition"]>

export type StoryTransitionSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  fromNodeId?: boolean
  pathId?: boolean
  toNodeId?: boolean
  ordering?: boolean
  condition?: boolean
  effect?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyTransition"]>

export type StoryTransitionSelectScalar = {
  id?: boolean
  storyId?: boolean
  fromNodeId?: boolean
  pathId?: boolean
  toNodeId?: boolean
  ordering?: boolean
  condition?: boolean
  effect?: boolean
  createdAt?: boolean
  updatedAt?: boolean
}

export type StoryTransitionOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "storyId" | "fromNodeId" | "pathId" | "toNodeId" | "ordering" | "condition" | "effect" | "createdAt" | "updatedAt", ExtArgs["result"]["storyTransition"]>
export type StoryTransitionInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}
export type StoryTransitionIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}
export type StoryTransitionIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  fromNode?: boolean | Prisma.StoryNodeDefaultArgs<ExtArgs>
  toNode?: boolean | Prisma.StoryTransition$toNodeArgs<ExtArgs>
  path?: boolean | Prisma.StoryPathDefaultArgs<ExtArgs>
}

export type $StoryTransitionPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "StoryTransition"
  objects: {
    story: Prisma.$TwineStoryPayload<ExtArgs>
    fromNode: Prisma.$StoryNodePayload<ExtArgs>
    toNode: Prisma.$StoryNodePayload<ExtArgs> | null
    path: Prisma.$StoryPathPayload<ExtArgs>
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: string
    storyId: string
    fromNodeId: string
    pathId: string
    toNodeId: string | null
    ordering: number | null
    condition: runtime.JsonValue | null
    effect: runtime.JsonValue | null
    createdAt: Date
    updatedAt: Date
  }, ExtArgs["result"]["storyTransition"]>
  composites: {}
}

export type StoryTransitionGetPayload<S extends boolean | null | undefined | StoryTransitionDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload, S>

export type StoryTransitionCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<StoryTransitionFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: StoryTransitionCountAggregateInputType | true
  }

export interface StoryTransitionDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['StoryTransition'], meta: { name: 'StoryTransition' } }
  /**
   * Find zero or one StoryTransition that matches the filter.
   * @param {StoryTransitionFindUniqueArgs} args - Arguments to find a StoryTransition
   * @example
   * // Get one StoryTransition
   * const storyTransition = await prisma.storyTransition.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends StoryTransitionFindUniqueArgs>(args: Prisma.SelectSubset<T, StoryTransitionFindUniqueArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one StoryTransition that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {StoryTransitionFindUniqueOrThrowArgs} args - Arguments to find a StoryTransition
   * @example
   * // Get one StoryTransition
   * const storyTransition = await prisma.storyTransition.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends StoryTransitionFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, StoryTransitionFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryTransition that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionFindFirstArgs} args - Arguments to find a StoryTransition
   * @example
   * // Get one StoryTransition
   * const storyTransition = await prisma.storyTransition.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends StoryTransitionFindFirstArgs>(args?: Prisma.SelectSubset<T, StoryTransitionFindFirstArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryTransition that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionFindFirstOrThrowArgs} args - Arguments to find a StoryTransition
   * @example
   * // Get one StoryTransition
   * const storyTransition = await prisma.storyTransition.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends StoryTransitionFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, StoryTransitionFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more StoryTransitions that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all StoryTransitions
   * const storyTransitions = await prisma.storyTransition.findMany()
   * 
   * // Get first 10 StoryTransitions
   * const storyTransitions = await prisma.storyTransition.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const storyTransitionWithIdOnly = await prisma.storyTransition.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends StoryTransitionFindManyArgs>(args?: Prisma.SelectSubset<T, StoryTransitionFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a StoryTransition.
   * @param {StoryTransitionCreateArgs} args - Arguments to create a StoryTransition.
   * @example
   * // Create one StoryTransition
   * const StoryTransition = await prisma.storyTransition.create({
   *   data: {
   *     // ... data to create a StoryTransition
   *   }
   * })
   * 
   */
  create<T extends StoryTransitionCreateArgs>(args: Prisma.SelectSubset<T, StoryTransitionCreateArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many StoryTransitions.
   * @param {StoryTransitionCreateManyArgs} args - Arguments to create many StoryTransitions.
   * @example
   * // Create many StoryTransitions
   * const storyTransition = await prisma.storyTransition.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends StoryTransitionCreateManyArgs>(args?: Prisma.SelectSubset<T, StoryTransitionCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many StoryTransitions and returns the data saved in the database.
   * @param {StoryTransitionCreateManyAndReturnArgs} args - Arguments to create many StoryTransitions.
   * @example
   * // Create many StoryTransitions
   * const storyTransition = await prisma.storyTransition.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many StoryTransitions and only return the `id`
   * const storyTransitionWithIdOnly = await prisma.storyTransition.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends StoryTransitionCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, StoryTransitionCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a StoryTransition.
   * @param {StoryTransitionDeleteArgs} args - Arguments to delete one StoryTransition.
   * @example
   * // Delete one StoryTransition
   * const StoryTransition = await prisma.storyTransition.delete({
   *   where: {
   *     // ... filter to delete one StoryTransition
   *   }
   * })
   * 
   */
  delete<T extends StoryTransitionDeleteArgs>(args: Prisma.SelectSubset<T, StoryTransitionDeleteArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one StoryTransition.
   * @param {StoryTransitionUpdateArgs} args - Arguments to update one StoryTransition.
   * @example
   * // Update one StoryTransition
   * const storyTransition = await prisma.storyTransition.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends StoryTransitionUpdateArgs>(args: Prisma.SelectSubset<T, StoryTransitionUpdateArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more StoryTransitions.
   * @param {StoryTransitionDeleteManyArgs} args - Arguments to filter StoryTransitions to delete.
   * @example
   * // Delete a few StoryTransitions
   * const { count } = await prisma.storyTransition.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends StoryTransitionDeleteManyArgs>(args?: Prisma.SelectSubset<T, StoryTransitionDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryTransitions.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many StoryTransitions
   * const storyTransition = await prisma.storyTransition.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends StoryTransitionUpdateManyArgs>(args: Prisma.SelectSubset<T, StoryTransitionUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryTransitions and returns the data updated in the database.
   * @param {StoryTransitionUpdateManyAndReturnArgs} args - Arguments to update many StoryTransitions.
   * @example
   * // Update many StoryTransitions
   * const storyTransition = await prisma.storyTransition.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more StoryTransitions and only return the `id`
   * const storyTransitionWithIdOnly = await prisma.storyTransition.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends StoryTransitionUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, StoryTransitionUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one StoryTransition.
   * @param {StoryTransitionUpsertArgs} args - Arguments to update or create a StoryTransition.
   * @example
   * // Update or create a StoryTransition
   * const storyTransition = await prisma.storyTransition.upsert({
   *   create: {
   *     // ... data to create a StoryTransition
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the StoryTransition we want to update
   *   }
   * })
   */
  upsert<T extends StoryTransitionUpsertArgs>(args: Prisma.SelectSubset<T, StoryTransitionUpsertArgs<ExtArgs>>): Prisma.Prisma__StoryTransitionClient<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of StoryTransitions.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionCountArgs} args - Arguments to filter StoryTransitions to count.
   * @example
   * // Count the number of StoryTransitions
   * const count = await prisma.storyTransition.count({
   *   where: {
   *     // ... the filter for the StoryTransitions we want to count
   *   }
   * })
  **/
  count<T extends StoryTransitionCountArgs>(
    args?: Prisma.Subset<T, StoryTransitionCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], StoryTransitionCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a StoryTransition.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends StoryTransitionAggregateArgs>(args: Prisma.Subset<T, StoryTransitionAggregateArgs>): Prisma.PrismaPromise<GetStoryTransitionAggregateType<T>>

  /**
   * Group by StoryTransition.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryTransitionGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends StoryTransitionGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: StoryTransitionGroupByArgs['orderBy'] }
      : { orderBy?: StoryTransitionGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, StoryTransitionGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetStoryTransitionGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the StoryTransition model
 */
readonly fields: StoryTransitionFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for StoryTransition.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__StoryTransitionClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  story<T extends Prisma.TwineStoryDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.TwineStoryDefaultArgs<ExtArgs>>): Prisma.Prisma__TwineStoryClient<runtime.Types.Result.GetResult<Prisma.$TwineStoryPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  fromNode<T extends Prisma.StoryNodeDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryNodeDefaultArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  toNode<T extends Prisma.StoryTransition$toNodeArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryTransition$toNodeArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>
  path<T extends Prisma.StoryPathDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryPathDefaultArgs<ExtArgs>>): Prisma.Prisma__StoryPathClient<runtime.Types.Result.GetResult<Prisma.$StoryPathPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the StoryTransition model
 */
export interface StoryTransitionFieldRefs {
  readonly id: Prisma.FieldRef<"StoryTransition", 'String'>
  readonly storyId: Prisma.FieldRef<"StoryTransition", 'String'>
  readonly fromNodeId: Prisma.FieldRef<"StoryTransition", 'String'>
  readonly pathId: Prisma.FieldRef<"StoryTransition", 'String'>
  readonly toNodeId: Prisma.FieldRef<"StoryTransition", 'String'>
  readonly ordering: Prisma.FieldRef<"StoryTransition", 'Int'>
  readonly condition: Prisma.FieldRef<"StoryTransition", 'Json'>
  readonly effect: Prisma.FieldRef<"StoryTransition", 'Json'>
  readonly createdAt: Prisma.FieldRef<"StoryTransition", 'DateTime'>
  readonly updatedAt: Prisma.FieldRef<"StoryTransition", 'DateTime'>
}
    

// Custom InputTypes
/**
 * StoryTransition findUnique
 */
export type StoryTransitionFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter, which StoryTransition to fetch.
   */
  where: Prisma.StoryTransitionWhereUniqueInput
}

/**
 * StoryTransition findUniqueOrThrow
 */
export type StoryTransitionFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter, which StoryTransition to fetch.
   */
  where: Prisma.StoryTransitionWhereUniqueInput
}

/**
 * StoryTransition findFirst
 */
export type StoryTransitionFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter, which StoryTransition to fetch.
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryTransitions to fetch.
   */
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryTransitions.
   */
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryTransitions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryTransitions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryTransitions.
   */
  distinct?: Prisma.StoryTransitionScalarFieldEnum | Prisma.StoryTransitionScalarFieldEnum[]
}

/**
 * StoryTransition findFirstOrThrow
 */
export type StoryTransitionFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter, which StoryTransition to fetch.
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryTransitions to fetch.
   */
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryTransitions.
   */
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryTransitions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryTransitions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryTransitions.
   */
  distinct?: Prisma.StoryTransitionScalarFieldEnum | Prisma.StoryTransitionScalarFieldEnum[]
}

/**
 * StoryTransition findMany
 */
export type StoryTransitionFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter, which StoryTransitions to fetch.
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryTransitions to fetch.
   */
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing StoryTransitions.
   */
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryTransitions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryTransitions.
   */
  skip?: number
  distinct?: Prisma.StoryTransitionScalarFieldEnum | Prisma.StoryTransitionScalarFieldEnum[]
}

/**
 * StoryTransition create
 */
export type StoryTransitionCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * The data needed to create a StoryTransition.
   */
  data: Prisma.XOR<Prisma.StoryTransitionCreateInput, Prisma.StoryTransitionUncheckedCreateInput>
}

/**
 * StoryTransition createMany
 */
export type StoryTransitionCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many StoryTransitions.
   */
  data: Prisma.StoryTransitionCreateManyInput | Prisma.StoryTransitionCreateManyInput[]
  skipDuplicates?: boolean
}

/**
 * StoryTransition createManyAndReturn
 */
export type StoryTransitionCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * The data used to create many StoryTransitions.
   */
  data: Prisma.StoryTransitionCreateManyInput | Prisma.StoryTransitionCreateManyInput[]
  skipDuplicates?: boolean
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * StoryTransition update
 */
export type StoryTransitionUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * The data needed to update a StoryTransition.
   */
  data: Prisma.XOR<Prisma.StoryTransitionUpdateInput, Prisma.StoryTransitionUncheckedUpdateInput>
  /**
   * Choose, which StoryTransition to update.
   */
  where: Prisma.StoryTransitionWhereUniqueInput
}

/**
 * StoryTransition updateMany
 */
export type StoryTransitionUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update StoryTransitions.
   */
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyInput>
  /**
   * Filter which StoryTransitions to update
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * Limit how many StoryTransitions to update.
   */
  limit?: number
}

/**
 * StoryTransition updateManyAndReturn
 */
export type StoryTransitionUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * The data used to update StoryTransitions.
   */
  data: Prisma.XOR<Prisma.StoryTransitionUpdateManyMutationInput, Prisma.StoryTransitionUncheckedUpdateManyInput>
  /**
   * Filter which StoryTransitions to update
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * Limit how many StoryTransitions to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * StoryTransition upsert
 */
export type StoryTransitionUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * The filter to search for the StoryTransition to update in case it exists.
   */
  where: Prisma.StoryTransitionWhereUniqueInput
  /**
   * In case the StoryTransition found by the `where` argument doesn't exist, create a new StoryTransition with this data.
   */
  create: Prisma.XOR<Prisma.StoryTransitionCreateInput, Prisma.StoryTransitionUncheckedCreateInput>
  /**
   * In case the StoryTransition was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.StoryTransitionUpdateInput, Prisma.StoryTransitionUncheckedUpdateInput>
}

/**
 * StoryTransition delete
 */
export type StoryTransitionDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  /**
   * Filter which StoryTransition to delete.
   */
  where: Prisma.StoryTransitionWhereUniqueInput
}

/**
 * StoryTransition deleteMany
 */
export type StoryTransitionDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryTransitions to delete
   */
  where?: Prisma.StoryTransitionWhereInput
  /**
   * Limit how many StoryTransitions to delete.
   */
  limit?: number
}

/**
 * StoryTransition.toNode
 */
export type StoryTransition$toNodeArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  where?: Prisma.StoryNodeWhereInput
}

/**
 * StoryTransition without action
 */
export type StoryTransitionDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
}
