// @prisma/client/models/StoryVersion.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// @ts-nocheck 
/*
 * This file exports the `StoryVersion` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/library"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model StoryVersion
 * 
 */
export type StoryVersionModel = runtime.Types.Result.DefaultSelection<Prisma.$StoryVersionPayload>

export type AggregateStoryVersion = {
  _count: StoryVersionCountAggregateOutputType | null
  _avg: StoryVersionAvgAggregateOutputType | null
  _sum: StoryVersionSumAggregateOutputType | null
  _min: StoryVersionMinAggregateOutputType | null
  _max: StoryVersionMaxAggregateOutputType | null
}

export type StoryVersionAvgAggregateOutputType = {
  versionNumber: number | null
}

export type StoryVersionSumAggregateOutputType = {
  versionNumber: number | null
}

export type StoryVersionMinAggregateOutputType = {
  id: string | null
  storyId: string | null
  authorId: string | null
  reviewerId: string | null
  versionNumber: number | null
  status: $Enums.StoryStatus | null
  changelog: string | null
  sourceUrl: string | null
  assetPath: string | null
  submittedAt: Date | null
  reviewedAt: Date | null
  ownershipStatus: $Enums.StoryOwnershipStatus | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryVersionMaxAggregateOutputType = {
  id: string | null
  storyId: string | null
  authorId: string | null
  reviewerId: string | null
  versionNumber: number | null
  status: $Enums.StoryStatus | null
  changelog: string | null
  sourceUrl: string | null
  assetPath: string | null
  submittedAt: Date | null
  reviewedAt: Date | null
  ownershipStatus: $Enums.StoryOwnershipStatus | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryVersionCountAggregateOutputType = {
  id: number
  storyId: number
  authorId: number
  reviewerId: number
  versionNumber: number
  status: number
  changelog: number
  content: number
  sourceUrl: number
  assetPath: number
  metadata: number
  submittedAt: number
  reviewedAt: number
  ownershipStatus: number
  consentSnapshot: number
  createdAt: number
  updatedAt: number
  _all: number
}


export type StoryVersionAvgAggregateInputType = {
  versionNumber?: true
}

export type StoryVersionSumAggregateInputType = {
  versionNumber?: true
}

export type StoryVersionMinAggregateInputType = {
  id?: true
  storyId?: true
  authorId?: true
  reviewerId?: true
  versionNumber?: true
  status?: true
  changelog?: true
  sourceUrl?: true
  assetPath?: true
  submittedAt?: true
  reviewedAt?: true
  ownershipStatus?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryVersionMaxAggregateInputType = {
  id?: true
  storyId?: true
  authorId?: true
  reviewerId?: true
  versionNumber?: true
  status?: true
  changelog?: true
  sourceUrl?: true
  assetPath?: true
  submittedAt?: true
  reviewedAt?: true
  ownershipStatus?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryVersionCountAggregateInputType = {
  id?: true
  storyId?: true
  authorId?: true
  reviewerId?: true
  versionNumber?: true
  status?: true
  changelog?: true
  content?: true
  sourceUrl?: true
  assetPath?: true
  metadata?: true
  submittedAt?: true
  reviewedAt?: true
  ownershipStatus?: true
  consentSnapshot?: true
  createdAt?: true
  updatedAt?: true
  _all?: true
}

export type StoryVersionAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryVersion to aggregate.
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryVersions to fetch.
   */
  orderBy?: Prisma.StoryVersionOrderByWithRelationInput | Prisma.StoryVersionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.StoryVersionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryVersions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryVersions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned StoryVersions
  **/
  _count?: true | StoryVersionCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: StoryVersionAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: StoryVersionSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: StoryVersionMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: StoryVersionMaxAggregateInputType
}

export type GetStoryVersionAggregateType<T extends StoryVersionAggregateArgs> = {
      [P in keyof T & keyof AggregateStoryVersion]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateStoryVersion[P]>
    : Prisma.GetScalarType<T[P], AggregateStoryVersion[P]>
}




export type StoryVersionGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryVersionWhereInput
  orderBy?: Prisma.StoryVersionOrderByWithAggregationInput | Prisma.StoryVersionOrderByWithAggregationInput[]
  by: Prisma.StoryVersionScalarFieldEnum[] | Prisma.StoryVersionScalarFieldEnum
  having?: Prisma.StoryVersionScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: StoryVersionCountAggregateInputType | true
  _avg?: StoryVersionAvgAggregateInputType
  _sum?: StoryVersionSumAggregateInputType
  _min?: StoryVersionMinAggregateInputType
  _max?: StoryVersionMaxAggregateInputType
}

export type StoryVersionGroupByOutputType = {
  id: string
  storyId: string
  authorId: string | null
  reviewerId: string | null
  versionNumber: number
  status: $Enums.StoryStatus
  changelog: string | null
  content: runtime.JsonValue | null
  sourceUrl: string | null
  assetPath: string | null
  metadata: runtime.JsonValue | null
  submittedAt: Date | null
  reviewedAt: Date | null
  ownershipStatus: $Enums.StoryOwnershipStatus
  consentSnapshot: runtime.JsonValue | null
  createdAt: Date
  updatedAt: Date | null
  _count: StoryVersionCountAggregateOutputType | null
  _avg: StoryVersionAvgAggregateOutputType | null
  _sum: StoryVersionSumAggregateOutputType | null
  _min: StoryVersionMinAggregateOutputType | null
  _max: StoryVersionMaxAggregateOutputType | null
}

type GetStoryVersionGroupByPayload<T extends StoryVersionGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<StoryVersionGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof StoryVersionGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], StoryVersionGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], StoryVersionGroupByOutputType[P]>
      }
    >
  >



export type StoryVersionWhereInput = {
  AND?: Prisma.StoryVersionWhereInput | Prisma.StoryVersionWhereInput[]
  OR?: Prisma.StoryVersionWhereInput[]
  NOT?: Prisma.StoryVersionWhereInput | Prisma.StoryVersionWhereInput[]
  id?: Prisma.StringFilter<"StoryVersion"> | string
  storyId?: Prisma.StringFilter<"StoryVersion"> | string
  authorId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  reviewerId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  versionNumber?: Prisma.IntFilter<"StoryVersion"> | number
  status?: Prisma.EnumStoryStatusFilter<"StoryVersion"> | $Enums.StoryStatus
  changelog?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  content?: Prisma.JsonNullableFilter<"StoryVersion">
  sourceUrl?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  assetPath?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  metadata?: Prisma.JsonNullableFilter<"StoryVersion">
  submittedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  reviewedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFilter<"StoryVersion"> | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.JsonNullableFilter<"StoryVersion">
  createdAt?: Prisma.DateTimeFilter<"StoryVersion"> | Date | string
  updatedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  featuredIn?: Prisma.XOR<Prisma.TwineStoryNullableScalarRelationFilter, Prisma.TwineStoryWhereInput> | null
  author?: Prisma.XOR<Prisma.UserNullableScalarRelationFilter, Prisma.UserWhereInput> | null
  reviewer?: Prisma.XOR<Prisma.UserNullableScalarRelationFilter, Prisma.UserWhereInput> | null
  reviews?: Prisma.StoryReviewListRelationFilter
  playSessions?: Prisma.StoryPlaySessionListRelationFilter
}

export type StoryVersionOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  authorId?: Prisma.SortOrderInput | Prisma.SortOrder
  reviewerId?: Prisma.SortOrderInput | Prisma.SortOrder
  versionNumber?: Prisma.SortOrder
  status?: Prisma.SortOrder
  changelog?: Prisma.SortOrderInput | Prisma.SortOrder
  content?: Prisma.SortOrderInput | Prisma.SortOrder
  sourceUrl?: Prisma.SortOrderInput | Prisma.SortOrder
  assetPath?: Prisma.SortOrderInput | Prisma.SortOrder
  metadata?: Prisma.SortOrderInput | Prisma.SortOrder
  submittedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  reviewedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  ownershipStatus?: Prisma.SortOrder
  consentSnapshot?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  story?: Prisma.TwineStoryOrderByWithRelationInput
  featuredIn?: Prisma.TwineStoryOrderByWithRelationInput
  author?: Prisma.UserOrderByWithRelationInput
  reviewer?: Prisma.UserOrderByWithRelationInput
  reviews?: Prisma.StoryReviewOrderByRelationAggregateInput
  playSessions?: Prisma.StoryPlaySessionOrderByRelationAggregateInput
}

export type StoryVersionWhereUniqueInput = Prisma.AtLeast<{
  id?: string
  AND?: Prisma.StoryVersionWhereInput | Prisma.StoryVersionWhereInput[]
  OR?: Prisma.StoryVersionWhereInput[]
  NOT?: Prisma.StoryVersionWhereInput | Prisma.StoryVersionWhereInput[]
  storyId?: Prisma.StringFilter<"StoryVersion"> | string
  authorId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  reviewerId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  versionNumber?: Prisma.IntFilter<"StoryVersion"> | number
  status?: Prisma.EnumStoryStatusFilter<"StoryVersion"> | $Enums.StoryStatus
  changelog?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  content?: Prisma.JsonNullableFilter<"StoryVersion">
  sourceUrl?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  assetPath?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  metadata?: Prisma.JsonNullableFilter<"StoryVersion">
  submittedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  reviewedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFilter<"StoryVersion"> | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.JsonNullableFilter<"StoryVersion">
  createdAt?: Prisma.DateTimeFilter<"StoryVersion"> | Date | string
  updatedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  featuredIn?: Prisma.XOR<Prisma.TwineStoryNullableScalarRelationFilter, Prisma.TwineStoryWhereInput> | null
  author?: Prisma.XOR<Prisma.UserNullableScalarRelationFilter, Prisma.UserWhereInput> | null
  reviewer?: Prisma.XOR<Prisma.UserNullableScalarRelationFilter, Prisma.UserWhereInput> | null
  reviews?: Prisma.StoryReviewListRelationFilter
  playSessions?: Prisma.StoryPlaySessionListRelationFilter
}, "id">

export type StoryVersionOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  authorId?: Prisma.SortOrderInput | Prisma.SortOrder
  reviewerId?: Prisma.SortOrderInput | Prisma.SortOrder
  versionNumber?: Prisma.SortOrder
  status?: Prisma.SortOrder
  changelog?: Prisma.SortOrderInput | Prisma.SortOrder
  content?: Prisma.SortOrderInput | Prisma.SortOrder
  sourceUrl?: Prisma.SortOrderInput | Prisma.SortOrder
  assetPath?: Prisma.SortOrderInput | Prisma.SortOrder
  metadata?: Prisma.SortOrderInput | Prisma.SortOrder
  submittedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  reviewedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  ownershipStatus?: Prisma.SortOrder
  consentSnapshot?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrderInput | Prisma.SortOrder
  _count?: Prisma.StoryVersionCountOrderByAggregateInput
  _avg?: Prisma.StoryVersionAvgOrderByAggregateInput
  _max?: Prisma.StoryVersionMaxOrderByAggregateInput
  _min?: Prisma.StoryVersionMinOrderByAggregateInput
  _sum?: Prisma.StoryVersionSumOrderByAggregateInput
}

export type StoryVersionScalarWhereWithAggregatesInput = {
  AND?: Prisma.StoryVersionScalarWhereWithAggregatesInput | Prisma.StoryVersionScalarWhereWithAggregatesInput[]
  OR?: Prisma.StoryVersionScalarWhereWithAggregatesInput[]
  NOT?: Prisma.StoryVersionScalarWhereWithAggregatesInput | Prisma.StoryVersionScalarWhereWithAggregatesInput[]
  id?: Prisma.StringWithAggregatesFilter<"StoryVersion"> | string
  storyId?: Prisma.StringWithAggregatesFilter<"StoryVersion"> | string
  authorId?: Prisma.StringNullableWithAggregatesFilter<"StoryVersion"> | string | null
  reviewerId?: Prisma.StringNullableWithAggregatesFilter<"StoryVersion"> | string | null
  versionNumber?: Prisma.IntWithAggregatesFilter<"StoryVersion"> | number
  status?: Prisma.EnumStoryStatusWithAggregatesFilter<"StoryVersion"> | $Enums.StoryStatus
  changelog?: Prisma.StringNullableWithAggregatesFilter<"StoryVersion"> | string | null
  content?: Prisma.JsonNullableWithAggregatesFilter<"StoryVersion">
  sourceUrl?: Prisma.StringNullableWithAggregatesFilter<"StoryVersion"> | string | null
  assetPath?: Prisma.StringNullableWithAggregatesFilter<"StoryVersion"> | string | null
  metadata?: Prisma.JsonNullableWithAggregatesFilter<"StoryVersion">
  submittedAt?: Prisma.DateTimeNullableWithAggregatesFilter<"StoryVersion"> | Date | string | null
  reviewedAt?: Prisma.DateTimeNullableWithAggregatesFilter<"StoryVersion"> | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusWithAggregatesFilter<"StoryVersion"> | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.JsonNullableWithAggregatesFilter<"StoryVersion">
  createdAt?: Prisma.DateTimeWithAggregatesFilter<"StoryVersion"> | Date | string
  updatedAt?: Prisma.DateTimeNullableWithAggregatesFilter<"StoryVersion"> | Date | string | null
}

export type StoryVersionCreateInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateInput = {
  id?: string
  storyId: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionCreateManyInput = {
  id?: string
  storyId: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
}

export type StoryVersionUpdateManyMutationInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
}

export type StoryVersionUncheckedUpdateManyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
}

export type StoryVersionListRelationFilter = {
  every?: Prisma.StoryVersionWhereInput
  some?: Prisma.StoryVersionWhereInput
  none?: Prisma.StoryVersionWhereInput
}

export type StoryVersionOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type StoryVersionNullableScalarRelationFilter = {
  is?: Prisma.StoryVersionWhereInput | null
  isNot?: Prisma.StoryVersionWhereInput | null
}

export type StoryVersionCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  authorId?: Prisma.SortOrder
  reviewerId?: Prisma.SortOrder
  versionNumber?: Prisma.SortOrder
  status?: Prisma.SortOrder
  changelog?: Prisma.SortOrder
  content?: Prisma.SortOrder
  sourceUrl?: Prisma.SortOrder
  assetPath?: Prisma.SortOrder
  metadata?: Prisma.SortOrder
  submittedAt?: Prisma.SortOrder
  reviewedAt?: Prisma.SortOrder
  ownershipStatus?: Prisma.SortOrder
  consentSnapshot?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryVersionAvgOrderByAggregateInput = {
  versionNumber?: Prisma.SortOrder
}

export type StoryVersionMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  authorId?: Prisma.SortOrder
  reviewerId?: Prisma.SortOrder
  versionNumber?: Prisma.SortOrder
  status?: Prisma.SortOrder
  changelog?: Prisma.SortOrder
  sourceUrl?: Prisma.SortOrder
  assetPath?: Prisma.SortOrder
  submittedAt?: Prisma.SortOrder
  reviewedAt?: Prisma.SortOrder
  ownershipStatus?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryVersionMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  authorId?: Prisma.SortOrder
  reviewerId?: Prisma.SortOrder
  versionNumber?: Prisma.SortOrder
  status?: Prisma.SortOrder
  changelog?: Prisma.SortOrder
  sourceUrl?: Prisma.SortOrder
  assetPath?: Prisma.SortOrder
  submittedAt?: Prisma.SortOrder
  reviewedAt?: Prisma.SortOrder
  ownershipStatus?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryVersionSumOrderByAggregateInput = {
  versionNumber?: Prisma.SortOrder
}

export type StoryVersionScalarRelationFilter = {
  is?: Prisma.StoryVersionWhereInput
  isNot?: Prisma.StoryVersionWhereInput
}

export type StoryVersionCreateNestedManyWithoutAuthorInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput> | Prisma.StoryVersionCreateWithoutAuthorInput[] | Prisma.StoryVersionUncheckedCreateWithoutAuthorInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutAuthorInput | Prisma.StoryVersionCreateOrConnectWithoutAuthorInput[]
  createMany?: Prisma.StoryVersionCreateManyAuthorInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionCreateNestedManyWithoutReviewerInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput> | Prisma.StoryVersionCreateWithoutReviewerInput[] | Prisma.StoryVersionUncheckedCreateWithoutReviewerInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewerInput | Prisma.StoryVersionCreateOrConnectWithoutReviewerInput[]
  createMany?: Prisma.StoryVersionCreateManyReviewerInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionUncheckedCreateNestedManyWithoutAuthorInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput> | Prisma.StoryVersionCreateWithoutAuthorInput[] | Prisma.StoryVersionUncheckedCreateWithoutAuthorInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutAuthorInput | Prisma.StoryVersionCreateOrConnectWithoutAuthorInput[]
  createMany?: Prisma.StoryVersionCreateManyAuthorInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionUncheckedCreateNestedManyWithoutReviewerInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput> | Prisma.StoryVersionCreateWithoutReviewerInput[] | Prisma.StoryVersionUncheckedCreateWithoutReviewerInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewerInput | Prisma.StoryVersionCreateOrConnectWithoutReviewerInput[]
  createMany?: Prisma.StoryVersionCreateManyReviewerInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionUpdateManyWithoutAuthorNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput> | Prisma.StoryVersionCreateWithoutAuthorInput[] | Prisma.StoryVersionUncheckedCreateWithoutAuthorInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutAuthorInput | Prisma.StoryVersionCreateOrConnectWithoutAuthorInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutAuthorInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutAuthorInput[]
  createMany?: Prisma.StoryVersionCreateManyAuthorInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutAuthorInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutAuthorInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutAuthorInput | Prisma.StoryVersionUpdateManyWithWhereWithoutAuthorInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type StoryVersionUpdateManyWithoutReviewerNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput> | Prisma.StoryVersionCreateWithoutReviewerInput[] | Prisma.StoryVersionUncheckedCreateWithoutReviewerInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewerInput | Prisma.StoryVersionCreateOrConnectWithoutReviewerInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutReviewerInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutReviewerInput[]
  createMany?: Prisma.StoryVersionCreateManyReviewerInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutReviewerInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutReviewerInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutReviewerInput | Prisma.StoryVersionUpdateManyWithWhereWithoutReviewerInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type StoryVersionUncheckedUpdateManyWithoutAuthorNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput> | Prisma.StoryVersionCreateWithoutAuthorInput[] | Prisma.StoryVersionUncheckedCreateWithoutAuthorInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutAuthorInput | Prisma.StoryVersionCreateOrConnectWithoutAuthorInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutAuthorInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutAuthorInput[]
  createMany?: Prisma.StoryVersionCreateManyAuthorInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutAuthorInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutAuthorInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutAuthorInput | Prisma.StoryVersionUpdateManyWithWhereWithoutAuthorInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type StoryVersionUncheckedUpdateManyWithoutReviewerNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput> | Prisma.StoryVersionCreateWithoutReviewerInput[] | Prisma.StoryVersionUncheckedCreateWithoutReviewerInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewerInput | Prisma.StoryVersionCreateOrConnectWithoutReviewerInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutReviewerInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutReviewerInput[]
  createMany?: Prisma.StoryVersionCreateManyReviewerInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutReviewerInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutReviewerInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutReviewerInput | Prisma.StoryVersionUpdateManyWithWhereWithoutReviewerInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type StoryVersionCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput> | Prisma.StoryVersionCreateWithoutStoryInput[] | Prisma.StoryVersionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutStoryInput | Prisma.StoryVersionCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryVersionCreateManyStoryInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionCreateNestedOneWithoutFeaturedInInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedCreateWithoutFeaturedInInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutFeaturedInInput
  connect?: Prisma.StoryVersionWhereUniqueInput
}

export type StoryVersionUncheckedCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput> | Prisma.StoryVersionCreateWithoutStoryInput[] | Prisma.StoryVersionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutStoryInput | Prisma.StoryVersionCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryVersionCreateManyStoryInputEnvelope
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
}

export type StoryVersionUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput> | Prisma.StoryVersionCreateWithoutStoryInput[] | Prisma.StoryVersionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutStoryInput | Prisma.StoryVersionCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryVersionCreateManyStoryInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutStoryInput | Prisma.StoryVersionUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type StoryVersionUpdateOneWithoutFeaturedInNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedCreateWithoutFeaturedInInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutFeaturedInInput
  upsert?: Prisma.StoryVersionUpsertWithoutFeaturedInInput
  disconnect?: Prisma.StoryVersionWhereInput | boolean
  delete?: Prisma.StoryVersionWhereInput | boolean
  connect?: Prisma.StoryVersionWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.StoryVersionUpdateToOneWithWhereWithoutFeaturedInInput, Prisma.StoryVersionUpdateWithoutFeaturedInInput>, Prisma.StoryVersionUncheckedUpdateWithoutFeaturedInInput>
}

export type StoryVersionUncheckedUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput> | Prisma.StoryVersionCreateWithoutStoryInput[] | Prisma.StoryVersionUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutStoryInput | Prisma.StoryVersionCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryVersionUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryVersionUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryVersionCreateManyStoryInputEnvelope
  set?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  disconnect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  delete?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  connect?: Prisma.StoryVersionWhereUniqueInput | Prisma.StoryVersionWhereUniqueInput[]
  update?: Prisma.StoryVersionUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryVersionUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryVersionUpdateManyWithWhereWithoutStoryInput | Prisma.StoryVersionUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
}

export type EnumStoryStatusFieldUpdateOperationsInput = {
  set?: $Enums.StoryStatus
}

export type StoryVersionCreateNestedOneWithoutReviewsInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewsInput, Prisma.StoryVersionUncheckedCreateWithoutReviewsInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewsInput
  connect?: Prisma.StoryVersionWhereUniqueInput
}

export type StoryVersionUpdateOneRequiredWithoutReviewsNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewsInput, Prisma.StoryVersionUncheckedCreateWithoutReviewsInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutReviewsInput
  upsert?: Prisma.StoryVersionUpsertWithoutReviewsInput
  connect?: Prisma.StoryVersionWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.StoryVersionUpdateToOneWithWhereWithoutReviewsInput, Prisma.StoryVersionUpdateWithoutReviewsInput>, Prisma.StoryVersionUncheckedUpdateWithoutReviewsInput>
}

export type StoryVersionCreateNestedOneWithoutPlaySessionsInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedCreateWithoutPlaySessionsInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutPlaySessionsInput
  connect?: Prisma.StoryVersionWhereUniqueInput
}

export type StoryVersionUpdateOneWithoutPlaySessionsNestedInput = {
  create?: Prisma.XOR<Prisma.StoryVersionCreateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedCreateWithoutPlaySessionsInput>
  connectOrCreate?: Prisma.StoryVersionCreateOrConnectWithoutPlaySessionsInput
  upsert?: Prisma.StoryVersionUpsertWithoutPlaySessionsInput
  disconnect?: Prisma.StoryVersionWhereInput | boolean
  delete?: Prisma.StoryVersionWhereInput | boolean
  connect?: Prisma.StoryVersionWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.StoryVersionUpdateToOneWithWhereWithoutPlaySessionsInput, Prisma.StoryVersionUpdateWithoutPlaySessionsInput>, Prisma.StoryVersionUncheckedUpdateWithoutPlaySessionsInput>
}

export type StoryVersionCreateWithoutAuthorInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutAuthorInput = {
  id?: string
  storyId: string
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutAuthorInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput>
}

export type StoryVersionCreateManyAuthorInputEnvelope = {
  data: Prisma.StoryVersionCreateManyAuthorInput | Prisma.StoryVersionCreateManyAuthorInput[]
  skipDuplicates?: boolean
}

export type StoryVersionCreateWithoutReviewerInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutReviewerInput = {
  id?: string
  storyId: string
  authorId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutReviewerInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput>
}

export type StoryVersionCreateManyReviewerInputEnvelope = {
  data: Prisma.StoryVersionCreateManyReviewerInput | Prisma.StoryVersionCreateManyReviewerInput[]
  skipDuplicates?: boolean
}

export type StoryVersionUpsertWithWhereUniqueWithoutAuthorInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutAuthorInput, Prisma.StoryVersionUncheckedUpdateWithoutAuthorInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutAuthorInput, Prisma.StoryVersionUncheckedCreateWithoutAuthorInput>
}

export type StoryVersionUpdateWithWhereUniqueWithoutAuthorInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutAuthorInput, Prisma.StoryVersionUncheckedUpdateWithoutAuthorInput>
}

export type StoryVersionUpdateManyWithWhereWithoutAuthorInput = {
  where: Prisma.StoryVersionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateManyMutationInput, Prisma.StoryVersionUncheckedUpdateManyWithoutAuthorInput>
}

export type StoryVersionScalarWhereInput = {
  AND?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
  OR?: Prisma.StoryVersionScalarWhereInput[]
  NOT?: Prisma.StoryVersionScalarWhereInput | Prisma.StoryVersionScalarWhereInput[]
  id?: Prisma.StringFilter<"StoryVersion"> | string
  storyId?: Prisma.StringFilter<"StoryVersion"> | string
  authorId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  reviewerId?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  versionNumber?: Prisma.IntFilter<"StoryVersion"> | number
  status?: Prisma.EnumStoryStatusFilter<"StoryVersion"> | $Enums.StoryStatus
  changelog?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  content?: Prisma.JsonNullableFilter<"StoryVersion">
  sourceUrl?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  assetPath?: Prisma.StringNullableFilter<"StoryVersion"> | string | null
  metadata?: Prisma.JsonNullableFilter<"StoryVersion">
  submittedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  reviewedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFilter<"StoryVersion"> | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.JsonNullableFilter<"StoryVersion">
  createdAt?: Prisma.DateTimeFilter<"StoryVersion"> | Date | string
  updatedAt?: Prisma.DateTimeNullableFilter<"StoryVersion"> | Date | string | null
}

export type StoryVersionUpsertWithWhereUniqueWithoutReviewerInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutReviewerInput, Prisma.StoryVersionUncheckedUpdateWithoutReviewerInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewerInput, Prisma.StoryVersionUncheckedCreateWithoutReviewerInput>
}

export type StoryVersionUpdateWithWhereUniqueWithoutReviewerInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutReviewerInput, Prisma.StoryVersionUncheckedUpdateWithoutReviewerInput>
}

export type StoryVersionUpdateManyWithWhereWithoutReviewerInput = {
  where: Prisma.StoryVersionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateManyMutationInput, Prisma.StoryVersionUncheckedUpdateManyWithoutReviewerInput>
}

export type StoryVersionCreateWithoutStoryInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutStoryInput = {
  id?: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutStoryInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput>
}

export type StoryVersionCreateManyStoryInputEnvelope = {
  data: Prisma.StoryVersionCreateManyStoryInput | Prisma.StoryVersionCreateManyStoryInput[]
  skipDuplicates?: boolean
}

export type StoryVersionCreateWithoutFeaturedInInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutFeaturedInInput = {
  id?: string
  storyId: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutFeaturedInInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedCreateWithoutFeaturedInInput>
}

export type StoryVersionUpsertWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutStoryInput, Prisma.StoryVersionUncheckedUpdateWithoutStoryInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutStoryInput, Prisma.StoryVersionUncheckedCreateWithoutStoryInput>
}

export type StoryVersionUpdateWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutStoryInput, Prisma.StoryVersionUncheckedUpdateWithoutStoryInput>
}

export type StoryVersionUpdateManyWithWhereWithoutStoryInput = {
  where: Prisma.StoryVersionScalarWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateManyMutationInput, Prisma.StoryVersionUncheckedUpdateManyWithoutStoryInput>
}

export type StoryVersionUpsertWithoutFeaturedInInput = {
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedUpdateWithoutFeaturedInInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedCreateWithoutFeaturedInInput>
  where?: Prisma.StoryVersionWhereInput
}

export type StoryVersionUpdateToOneWithWhereWithoutFeaturedInInput = {
  where?: Prisma.StoryVersionWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutFeaturedInInput, Prisma.StoryVersionUncheckedUpdateWithoutFeaturedInInput>
}

export type StoryVersionUpdateWithoutFeaturedInInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutFeaturedInInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionCreateWithoutReviewsInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  playSessions?: Prisma.StoryPlaySessionCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutReviewsInput = {
  id?: string
  storyId: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  playSessions?: Prisma.StoryPlaySessionUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutReviewsInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewsInput, Prisma.StoryVersionUncheckedCreateWithoutReviewsInput>
}

export type StoryVersionUpsertWithoutReviewsInput = {
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutReviewsInput, Prisma.StoryVersionUncheckedUpdateWithoutReviewsInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutReviewsInput, Prisma.StoryVersionUncheckedCreateWithoutReviewsInput>
  where?: Prisma.StoryVersionWhereInput
}

export type StoryVersionUpdateToOneWithWhereWithoutReviewsInput = {
  where?: Prisma.StoryVersionWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutReviewsInput, Prisma.StoryVersionUncheckedUpdateWithoutReviewsInput>
}

export type StoryVersionUpdateWithoutReviewsInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutReviewsInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionCreateWithoutPlaySessionsInput = {
  id?: string
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  story: Prisma.TwineStoryCreateNestedOneWithoutVersionsInput
  featuredIn?: Prisma.TwineStoryCreateNestedOneWithoutLatestVersionInput
  author?: Prisma.UserCreateNestedOneWithoutAuthoredVersionsInput
  reviewer?: Prisma.UserCreateNestedOneWithoutReviewedVersionsInput
  reviews?: Prisma.StoryReviewCreateNestedManyWithoutVersionInput
}

export type StoryVersionUncheckedCreateWithoutPlaySessionsInput = {
  id?: string
  storyId: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedCreateNestedOneWithoutLatestVersionInput
  reviews?: Prisma.StoryReviewUncheckedCreateNestedManyWithoutVersionInput
}

export type StoryVersionCreateOrConnectWithoutPlaySessionsInput = {
  where: Prisma.StoryVersionWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedCreateWithoutPlaySessionsInput>
}

export type StoryVersionUpsertWithoutPlaySessionsInput = {
  update: Prisma.XOR<Prisma.StoryVersionUpdateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedUpdateWithoutPlaySessionsInput>
  create: Prisma.XOR<Prisma.StoryVersionCreateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedCreateWithoutPlaySessionsInput>
  where?: Prisma.StoryVersionWhereInput
}

export type StoryVersionUpdateToOneWithWhereWithoutPlaySessionsInput = {
  where?: Prisma.StoryVersionWhereInput
  data: Prisma.XOR<Prisma.StoryVersionUpdateWithoutPlaySessionsInput, Prisma.StoryVersionUncheckedUpdateWithoutPlaySessionsInput>
}

export type StoryVersionUpdateWithoutPlaySessionsInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutPlaySessionsInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionCreateManyAuthorInput = {
  id?: string
  storyId: string
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
}

export type StoryVersionCreateManyReviewerInput = {
  id?: string
  storyId: string
  authorId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
}

export type StoryVersionUpdateWithoutAuthorInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutAuthorInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateManyWithoutAuthorInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
}

export type StoryVersionUpdateWithoutReviewerInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutVersionsNestedInput
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutReviewerInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateManyWithoutReviewerInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
}

export type StoryVersionCreateManyStoryInput = {
  id?: string
  authorId?: string | null
  reviewerId?: string | null
  versionNumber: number
  status?: $Enums.StoryStatus
  changelog?: string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: string | null
  assetPath?: string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Date | string | null
  reviewedAt?: Date | string | null
  ownershipStatus?: $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string | null
}

export type StoryVersionUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUpdateOneWithoutLatestVersionNestedInput
  author?: Prisma.UserUpdateOneWithoutAuthoredVersionsNestedInput
  reviewer?: Prisma.UserUpdateOneWithoutReviewedVersionsNestedInput
  reviews?: Prisma.StoryReviewUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  featuredIn?: Prisma.TwineStoryUncheckedUpdateOneWithoutLatestVersionNestedInput
  reviews?: Prisma.StoryReviewUncheckedUpdateManyWithoutVersionNestedInput
  playSessions?: Prisma.StoryPlaySessionUncheckedUpdateManyWithoutVersionNestedInput
}

export type StoryVersionUncheckedUpdateManyWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  authorId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  reviewerId?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  versionNumber?: Prisma.IntFieldUpdateOperationsInput | number
  status?: Prisma.EnumStoryStatusFieldUpdateOperationsInput | $Enums.StoryStatus
  changelog?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  sourceUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  assetPath?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  metadata?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  submittedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  reviewedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  ownershipStatus?: Prisma.EnumStoryOwnershipStatusFieldUpdateOperationsInput | $Enums.StoryOwnershipStatus
  consentSnapshot?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
}


/**
 * Count Type StoryVersionCountOutputType
 */

export type StoryVersionCountOutputType = {
  reviews: number
  playSessions: number
}

export type StoryVersionCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  reviews?: boolean | StoryVersionCountOutputTypeCountReviewsArgs
  playSessions?: boolean | StoryVersionCountOutputTypeCountPlaySessionsArgs
}

/**
 * StoryVersionCountOutputType without action
 */
export type StoryVersionCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersionCountOutputType
   */
  select?: Prisma.StoryVersionCountOutputTypeSelect<ExtArgs> | null
}

/**
 * StoryVersionCountOutputType without action
 */
export type StoryVersionCountOutputTypeCountReviewsArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryReviewWhereInput
}

/**
 * StoryVersionCountOutputType without action
 */
export type StoryVersionCountOutputTypeCountPlaySessionsArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryPlaySessionWhereInput
}


export type StoryVersionSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  authorId?: boolean
  reviewerId?: boolean
  versionNumber?: boolean
  status?: boolean
  changelog?: boolean
  content?: boolean
  sourceUrl?: boolean
  assetPath?: boolean
  metadata?: boolean
  submittedAt?: boolean
  reviewedAt?: boolean
  ownershipStatus?: boolean
  consentSnapshot?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  featuredIn?: boolean | Prisma.StoryVersion$featuredInArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
  reviews?: boolean | Prisma.StoryVersion$reviewsArgs<ExtArgs>
  playSessions?: boolean | Prisma.StoryVersion$playSessionsArgs<ExtArgs>
  _count?: boolean | Prisma.StoryVersionCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyVersion"]>

export type StoryVersionSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  authorId?: boolean
  reviewerId?: boolean
  versionNumber?: boolean
  status?: boolean
  changelog?: boolean
  content?: boolean
  sourceUrl?: boolean
  assetPath?: boolean
  metadata?: boolean
  submittedAt?: boolean
  reviewedAt?: boolean
  ownershipStatus?: boolean
  consentSnapshot?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
}, ExtArgs["result"]["storyVersion"]>

export type StoryVersionSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  authorId?: boolean
  reviewerId?: boolean
  versionNumber?: boolean
  status?: boolean
  changelog?: boolean
  content?: boolean
  sourceUrl?: boolean
  assetPath?: boolean
  metadata?: boolean
  submittedAt?: boolean
  reviewedAt?: boolean
  ownershipStatus?: boolean
  consentSnapshot?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
}, ExtArgs["result"]["storyVersion"]>

export type StoryVersionSelectScalar = {
  id?: boolean
  storyId?: boolean
  authorId?: boolean
  reviewerId?: boolean
  versionNumber?: boolean
  status?: boolean
  changelog?: boolean
  content?: boolean
  sourceUrl?: boolean
  assetPath?: boolean
  metadata?: boolean
  submittedAt?: boolean
  reviewedAt?: boolean
  ownershipStatus?: boolean
  consentSnapshot?: boolean
  createdAt?: boolean
  updatedAt?: boolean
}

export type StoryVersionOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "storyId" | "authorId" | "reviewerId" | "versionNumber" | "status" | "changelog" | "content" | "sourceUrl" | "assetPath" | "metadata" | "submittedAt" | "reviewedAt" | "ownershipStatus" | "consentSnapshot" | "createdAt" | "updatedAt", ExtArgs["result"]["storyVersion"]>
export type StoryVersionInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  featuredIn?: boolean | Prisma.StoryVersion$featuredInArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
  reviews?: boolean | Prisma.StoryVersion$reviewsArgs<ExtArgs>
  playSessions?: boolean | Prisma.StoryVersion$playSessionsArgs<ExtArgs>
  _count?: boolean | Prisma.StoryVersionCountOutputTypeDefaultArgs<ExtArgs>
}
export type StoryVersionIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
}
export type StoryVersionIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  author?: boolean | Prisma.StoryVersion$authorArgs<ExtArgs>
  reviewer?: boolean | Prisma.StoryVersion$reviewerArgs<ExtArgs>
}

export type $StoryVersionPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "StoryVersion"
  objects: {
    story: Prisma.$TwineStoryPayload<ExtArgs>
    featuredIn: Prisma.$TwineStoryPayload<ExtArgs> | null
    author: Prisma.$UserPayload<ExtArgs> | null
    reviewer: Prisma.$UserPayload<ExtArgs> | null
    reviews: Prisma.$StoryReviewPayload<ExtArgs>[]
    playSessions: Prisma.$StoryPlaySessionPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: string
    storyId: string
    authorId: string | null
    reviewerId: string | null
    versionNumber: number
    status: $Enums.StoryStatus
    changelog: string | null
    content: runtime.JsonValue | null
    sourceUrl: string | null
    assetPath: string | null
    metadata: runtime.JsonValue | null
    submittedAt: Date | null
    reviewedAt: Date | null
    ownershipStatus: $Enums.StoryOwnershipStatus
    consentSnapshot: runtime.JsonValue | null
    createdAt: Date
    updatedAt: Date | null
  }, ExtArgs["result"]["storyVersion"]>
  composites: {}
}

export type StoryVersionGetPayload<S extends boolean | null | undefined | StoryVersionDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload, S>

export type StoryVersionCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<StoryVersionFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: StoryVersionCountAggregateInputType | true
  }

export interface StoryVersionDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['StoryVersion'], meta: { name: 'StoryVersion' } }
  /**
   * Find zero or one StoryVersion that matches the filter.
   * @param {StoryVersionFindUniqueArgs} args - Arguments to find a StoryVersion
   * @example
   * // Get one StoryVersion
   * const storyVersion = await prisma.storyVersion.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends StoryVersionFindUniqueArgs>(args: Prisma.SelectSubset<T, StoryVersionFindUniqueArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one StoryVersion that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {StoryVersionFindUniqueOrThrowArgs} args - Arguments to find a StoryVersion
   * @example
   * // Get one StoryVersion
   * const storyVersion = await prisma.storyVersion.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends StoryVersionFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, StoryVersionFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryVersion that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionFindFirstArgs} args - Arguments to find a StoryVersion
   * @example
   * // Get one StoryVersion
   * const storyVersion = await prisma.storyVersion.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends StoryVersionFindFirstArgs>(args?: Prisma.SelectSubset<T, StoryVersionFindFirstArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryVersion that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionFindFirstOrThrowArgs} args - Arguments to find a StoryVersion
   * @example
   * // Get one StoryVersion
   * const storyVersion = await prisma.storyVersion.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends StoryVersionFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, StoryVersionFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more StoryVersions that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all StoryVersions
   * const storyVersions = await prisma.storyVersion.findMany()
   * 
   * // Get first 10 StoryVersions
   * const storyVersions = await prisma.storyVersion.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const storyVersionWithIdOnly = await prisma.storyVersion.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends StoryVersionFindManyArgs>(args?: Prisma.SelectSubset<T, StoryVersionFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a StoryVersion.
   * @param {StoryVersionCreateArgs} args - Arguments to create a StoryVersion.
   * @example
   * // Create one StoryVersion
   * const StoryVersion = await prisma.storyVersion.create({
   *   data: {
   *     // ... data to create a StoryVersion
   *   }
   * })
   * 
   */
  create<T extends StoryVersionCreateArgs>(args: Prisma.SelectSubset<T, StoryVersionCreateArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many StoryVersions.
   * @param {StoryVersionCreateManyArgs} args - Arguments to create many StoryVersions.
   * @example
   * // Create many StoryVersions
   * const storyVersion = await prisma.storyVersion.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends StoryVersionCreateManyArgs>(args?: Prisma.SelectSubset<T, StoryVersionCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many StoryVersions and returns the data saved in the database.
   * @param {StoryVersionCreateManyAndReturnArgs} args - Arguments to create many StoryVersions.
   * @example
   * // Create many StoryVersions
   * const storyVersion = await prisma.storyVersion.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many StoryVersions and only return the `id`
   * const storyVersionWithIdOnly = await prisma.storyVersion.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends StoryVersionCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, StoryVersionCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a StoryVersion.
   * @param {StoryVersionDeleteArgs} args - Arguments to delete one StoryVersion.
   * @example
   * // Delete one StoryVersion
   * const StoryVersion = await prisma.storyVersion.delete({
   *   where: {
   *     // ... filter to delete one StoryVersion
   *   }
   * })
   * 
   */
  delete<T extends StoryVersionDeleteArgs>(args: Prisma.SelectSubset<T, StoryVersionDeleteArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one StoryVersion.
   * @param {StoryVersionUpdateArgs} args - Arguments to update one StoryVersion.
   * @example
   * // Update one StoryVersion
   * const storyVersion = await prisma.storyVersion.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends StoryVersionUpdateArgs>(args: Prisma.SelectSubset<T, StoryVersionUpdateArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more StoryVersions.
   * @param {StoryVersionDeleteManyArgs} args - Arguments to filter StoryVersions to delete.
   * @example
   * // Delete a few StoryVersions
   * const { count } = await prisma.storyVersion.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends StoryVersionDeleteManyArgs>(args?: Prisma.SelectSubset<T, StoryVersionDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryVersions.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many StoryVersions
   * const storyVersion = await prisma.storyVersion.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends StoryVersionUpdateManyArgs>(args: Prisma.SelectSubset<T, StoryVersionUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryVersions and returns the data updated in the database.
   * @param {StoryVersionUpdateManyAndReturnArgs} args - Arguments to update many StoryVersions.
   * @example
   * // Update many StoryVersions
   * const storyVersion = await prisma.storyVersion.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more StoryVersions and only return the `id`
   * const storyVersionWithIdOnly = await prisma.storyVersion.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends StoryVersionUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, StoryVersionUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one StoryVersion.
   * @param {StoryVersionUpsertArgs} args - Arguments to update or create a StoryVersion.
   * @example
   * // Update or create a StoryVersion
   * const storyVersion = await prisma.storyVersion.upsert({
   *   create: {
   *     // ... data to create a StoryVersion
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the StoryVersion we want to update
   *   }
   * })
   */
  upsert<T extends StoryVersionUpsertArgs>(args: Prisma.SelectSubset<T, StoryVersionUpsertArgs<ExtArgs>>): Prisma.Prisma__StoryVersionClient<runtime.Types.Result.GetResult<Prisma.$StoryVersionPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of StoryVersions.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionCountArgs} args - Arguments to filter StoryVersions to count.
   * @example
   * // Count the number of StoryVersions
   * const count = await prisma.storyVersion.count({
   *   where: {
   *     // ... the filter for the StoryVersions we want to count
   *   }
   * })
  **/
  count<T extends StoryVersionCountArgs>(
    args?: Prisma.Subset<T, StoryVersionCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], StoryVersionCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a StoryVersion.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends StoryVersionAggregateArgs>(args: Prisma.Subset<T, StoryVersionAggregateArgs>): Prisma.PrismaPromise<GetStoryVersionAggregateType<T>>

  /**
   * Group by StoryVersion.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryVersionGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends StoryVersionGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: StoryVersionGroupByArgs['orderBy'] }
      : { orderBy?: StoryVersionGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, StoryVersionGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetStoryVersionGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the StoryVersion model
 */
readonly fields: StoryVersionFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for StoryVersion.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__StoryVersionClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  story<T extends Prisma.TwineStoryDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.TwineStoryDefaultArgs<ExtArgs>>): Prisma.Prisma__TwineStoryClient<runtime.Types.Result.GetResult<Prisma.$TwineStoryPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  featuredIn<T extends Prisma.StoryVersion$featuredInArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryVersion$featuredInArgs<ExtArgs>>): Prisma.Prisma__TwineStoryClient<runtime.Types.Result.GetResult<Prisma.$TwineStoryPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>
  author<T extends Prisma.StoryVersion$authorArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryVersion$authorArgs<ExtArgs>>): Prisma.Prisma__UserClient<runtime.Types.Result.GetResult<Prisma.$UserPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>
  reviewer<T extends Prisma.StoryVersion$reviewerArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryVersion$reviewerArgs<ExtArgs>>): Prisma.Prisma__UserClient<runtime.Types.Result.GetResult<Prisma.$UserPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>
  reviews<T extends Prisma.StoryVersion$reviewsArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryVersion$reviewsArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryReviewPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  playSessions<T extends Prisma.StoryVersion$playSessionsArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryVersion$playSessionsArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryPlaySessionPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the StoryVersion model
 */
export interface StoryVersionFieldRefs {
  readonly id: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly storyId: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly authorId: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly reviewerId: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly versionNumber: Prisma.FieldRef<"StoryVersion", 'Int'>
  readonly status: Prisma.FieldRef<"StoryVersion", 'StoryStatus'>
  readonly changelog: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly content: Prisma.FieldRef<"StoryVersion", 'Json'>
  readonly sourceUrl: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly assetPath: Prisma.FieldRef<"StoryVersion", 'String'>
  readonly metadata: Prisma.FieldRef<"StoryVersion", 'Json'>
  readonly submittedAt: Prisma.FieldRef<"StoryVersion", 'DateTime'>
  readonly reviewedAt: Prisma.FieldRef<"StoryVersion", 'DateTime'>
  readonly ownershipStatus: Prisma.FieldRef<"StoryVersion", 'StoryOwnershipStatus'>
  readonly consentSnapshot: Prisma.FieldRef<"StoryVersion", 'Json'>
  readonly createdAt: Prisma.FieldRef<"StoryVersion", 'DateTime'>
  readonly updatedAt: Prisma.FieldRef<"StoryVersion", 'DateTime'>
}
    

// Custom InputTypes
/**
 * StoryVersion findUnique
 */
export type StoryVersionFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter, which StoryVersion to fetch.
   */
  where: Prisma.StoryVersionWhereUniqueInput
}

/**
 * StoryVersion findUniqueOrThrow
 */
export type StoryVersionFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter, which StoryVersion to fetch.
   */
  where: Prisma.StoryVersionWhereUniqueInput
}

/**
 * StoryVersion findFirst
 */
export type StoryVersionFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter, which StoryVersion to fetch.
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryVersions to fetch.
   */
  orderBy?: Prisma.StoryVersionOrderByWithRelationInput | Prisma.StoryVersionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryVersions.
   */
  cursor?: Prisma.StoryVersionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryVersions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryVersions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryVersions.
   */
  distinct?: Prisma.StoryVersionScalarFieldEnum | Prisma.StoryVersionScalarFieldEnum[]
}

/**
 * StoryVersion findFirstOrThrow
 */
export type StoryVersionFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter, which StoryVersion to fetch.
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryVersions to fetch.
   */
  orderBy?: Prisma.StoryVersionOrderByWithRelationInput | Prisma.StoryVersionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryVersions.
   */
  cursor?: Prisma.StoryVersionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryVersions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryVersions.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryVersions.
   */
  distinct?: Prisma.StoryVersionScalarFieldEnum | Prisma.StoryVersionScalarFieldEnum[]
}

/**
 * StoryVersion findMany
 */
export type StoryVersionFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter, which StoryVersions to fetch.
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryVersions to fetch.
   */
  orderBy?: Prisma.StoryVersionOrderByWithRelationInput | Prisma.StoryVersionOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing StoryVersions.
   */
  cursor?: Prisma.StoryVersionWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryVersions from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryVersions.
   */
  skip?: number
  distinct?: Prisma.StoryVersionScalarFieldEnum | Prisma.StoryVersionScalarFieldEnum[]
}

/**
 * StoryVersion create
 */
export type StoryVersionCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * The data needed to create a StoryVersion.
   */
  data: Prisma.XOR<Prisma.StoryVersionCreateInput, Prisma.StoryVersionUncheckedCreateInput>
}

/**
 * StoryVersion createMany
 */
export type StoryVersionCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many StoryVersions.
   */
  data: Prisma.StoryVersionCreateManyInput | Prisma.StoryVersionCreateManyInput[]
  skipDuplicates?: boolean
}

/**
 * StoryVersion createManyAndReturn
 */
export type StoryVersionCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * The data used to create many StoryVersions.
   */
  data: Prisma.StoryVersionCreateManyInput | Prisma.StoryVersionCreateManyInput[]
  skipDuplicates?: boolean
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * StoryVersion update
 */
export type StoryVersionUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * The data needed to update a StoryVersion.
   */
  data: Prisma.XOR<Prisma.StoryVersionUpdateInput, Prisma.StoryVersionUncheckedUpdateInput>
  /**
   * Choose, which StoryVersion to update.
   */
  where: Prisma.StoryVersionWhereUniqueInput
}

/**
 * StoryVersion updateMany
 */
export type StoryVersionUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update StoryVersions.
   */
  data: Prisma.XOR<Prisma.StoryVersionUpdateManyMutationInput, Prisma.StoryVersionUncheckedUpdateManyInput>
  /**
   * Filter which StoryVersions to update
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * Limit how many StoryVersions to update.
   */
  limit?: number
}

/**
 * StoryVersion updateManyAndReturn
 */
export type StoryVersionUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * The data used to update StoryVersions.
   */
  data: Prisma.XOR<Prisma.StoryVersionUpdateManyMutationInput, Prisma.StoryVersionUncheckedUpdateManyInput>
  /**
   * Filter which StoryVersions to update
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * Limit how many StoryVersions to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * StoryVersion upsert
 */
export type StoryVersionUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * The filter to search for the StoryVersion to update in case it exists.
   */
  where: Prisma.StoryVersionWhereUniqueInput
  /**
   * In case the StoryVersion found by the `where` argument doesn't exist, create a new StoryVersion with this data.
   */
  create: Prisma.XOR<Prisma.StoryVersionCreateInput, Prisma.StoryVersionUncheckedCreateInput>
  /**
   * In case the StoryVersion was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.StoryVersionUpdateInput, Prisma.StoryVersionUncheckedUpdateInput>
}

/**
 * StoryVersion delete
 */
export type StoryVersionDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
  /**
   * Filter which StoryVersion to delete.
   */
  where: Prisma.StoryVersionWhereUniqueInput
}

/**
 * StoryVersion deleteMany
 */
export type StoryVersionDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryVersions to delete
   */
  where?: Prisma.StoryVersionWhereInput
  /**
   * Limit how many StoryVersions to delete.
   */
  limit?: number
}

/**
 * StoryVersion.featuredIn
 */
export type StoryVersion$featuredInArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the TwineStory
   */
  select?: Prisma.TwineStorySelect<ExtArgs> | null
  /**
   * Omit specific fields from the TwineStory
   */
  omit?: Prisma.TwineStoryOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.TwineStoryInclude<ExtArgs> | null
  where?: Prisma.TwineStoryWhereInput
}

/**
 * StoryVersion.author
 */
export type StoryVersion$authorArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the User
   */
  select?: Prisma.UserSelect<ExtArgs> | null
  /**
   * Omit specific fields from the User
   */
  omit?: Prisma.UserOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.UserInclude<ExtArgs> | null
  where?: Prisma.UserWhereInput
}

/**
 * StoryVersion.reviewer
 */
export type StoryVersion$reviewerArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the User
   */
  select?: Prisma.UserSelect<ExtArgs> | null
  /**
   * Omit specific fields from the User
   */
  omit?: Prisma.UserOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.UserInclude<ExtArgs> | null
  where?: Prisma.UserWhereInput
}

/**
 * StoryVersion.reviews
 */
export type StoryVersion$reviewsArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryReview
   */
  select?: Prisma.StoryReviewSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryReview
   */
  omit?: Prisma.StoryReviewOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryReviewInclude<ExtArgs> | null
  where?: Prisma.StoryReviewWhereInput
  orderBy?: Prisma.StoryReviewOrderByWithRelationInput | Prisma.StoryReviewOrderByWithRelationInput[]
  cursor?: Prisma.StoryReviewWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.StoryReviewScalarFieldEnum | Prisma.StoryReviewScalarFieldEnum[]
}

/**
 * StoryVersion.playSessions
 */
export type StoryVersion$playSessionsArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryPlaySession
   */
  select?: Prisma.StoryPlaySessionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryPlaySession
   */
  omit?: Prisma.StoryPlaySessionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryPlaySessionInclude<ExtArgs> | null
  where?: Prisma.StoryPlaySessionWhereInput
  orderBy?: Prisma.StoryPlaySessionOrderByWithRelationInput | Prisma.StoryPlaySessionOrderByWithRelationInput[]
  cursor?: Prisma.StoryPlaySessionWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.StoryPlaySessionScalarFieldEnum | Prisma.StoryPlaySessionScalarFieldEnum[]
}

/**
 * StoryVersion without action
 */
export type StoryVersionDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryVersion
   */
  select?: Prisma.StoryVersionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryVersion
   */
  omit?: Prisma.StoryVersionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryVersionInclude<ExtArgs> | null
}
