// @prisma/client/models/StoryNode.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// @ts-nocheck 
/*
 * This file exports the `StoryNode` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/library"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model StoryNode
 * 
 */
export type StoryNodeModel = runtime.Types.Result.DefaultSelection<Prisma.$StoryNodePayload>

export type AggregateStoryNode = {
  _count: StoryNodeCountAggregateOutputType | null
  _min: StoryNodeMinAggregateOutputType | null
  _max: StoryNodeMaxAggregateOutputType | null
}

export type StoryNodeMinAggregateOutputType = {
  id: string | null
  storyId: string | null
  key: string | null
  title: string | null
  synopsis: string | null
  type: $Enums.StoryNodeType | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryNodeMaxAggregateOutputType = {
  id: string | null
  storyId: string | null
  key: string | null
  title: string | null
  synopsis: string | null
  type: $Enums.StoryNodeType | null
  createdAt: Date | null
  updatedAt: Date | null
}

export type StoryNodeCountAggregateOutputType = {
  id: number
  storyId: number
  key: number
  title: number
  synopsis: number
  type: number
  content: number
  media: number
  createdAt: number
  updatedAt: number
  _all: number
}


export type StoryNodeMinAggregateInputType = {
  id?: true
  storyId?: true
  key?: true
  title?: true
  synopsis?: true
  type?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryNodeMaxAggregateInputType = {
  id?: true
  storyId?: true
  key?: true
  title?: true
  synopsis?: true
  type?: true
  createdAt?: true
  updatedAt?: true
}

export type StoryNodeCountAggregateInputType = {
  id?: true
  storyId?: true
  key?: true
  title?: true
  synopsis?: true
  type?: true
  content?: true
  media?: true
  createdAt?: true
  updatedAt?: true
  _all?: true
}

export type StoryNodeAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryNode to aggregate.
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryNodes to fetch.
   */
  orderBy?: Prisma.StoryNodeOrderByWithRelationInput | Prisma.StoryNodeOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.StoryNodeWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryNodes from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryNodes.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned StoryNodes
  **/
  _count?: true | StoryNodeCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: StoryNodeMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: StoryNodeMaxAggregateInputType
}

export type GetStoryNodeAggregateType<T extends StoryNodeAggregateArgs> = {
      [P in keyof T & keyof AggregateStoryNode]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateStoryNode[P]>
    : Prisma.GetScalarType<T[P], AggregateStoryNode[P]>
}




export type StoryNodeGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryNodeWhereInput
  orderBy?: Prisma.StoryNodeOrderByWithAggregationInput | Prisma.StoryNodeOrderByWithAggregationInput[]
  by: Prisma.StoryNodeScalarFieldEnum[] | Prisma.StoryNodeScalarFieldEnum
  having?: Prisma.StoryNodeScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: StoryNodeCountAggregateInputType | true
  _min?: StoryNodeMinAggregateInputType
  _max?: StoryNodeMaxAggregateInputType
}

export type StoryNodeGroupByOutputType = {
  id: string
  storyId: string
  key: string
  title: string | null
  synopsis: string | null
  type: $Enums.StoryNodeType
  content: runtime.JsonValue | null
  media: runtime.JsonValue | null
  createdAt: Date
  updatedAt: Date
  _count: StoryNodeCountAggregateOutputType | null
  _min: StoryNodeMinAggregateOutputType | null
  _max: StoryNodeMaxAggregateOutputType | null
}

type GetStoryNodeGroupByPayload<T extends StoryNodeGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<StoryNodeGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof StoryNodeGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], StoryNodeGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], StoryNodeGroupByOutputType[P]>
      }
    >
  >



export type StoryNodeWhereInput = {
  AND?: Prisma.StoryNodeWhereInput | Prisma.StoryNodeWhereInput[]
  OR?: Prisma.StoryNodeWhereInput[]
  NOT?: Prisma.StoryNodeWhereInput | Prisma.StoryNodeWhereInput[]
  id?: Prisma.StringFilter<"StoryNode"> | string
  storyId?: Prisma.StringFilter<"StoryNode"> | string
  key?: Prisma.StringFilter<"StoryNode"> | string
  title?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  synopsis?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  type?: Prisma.EnumStoryNodeTypeFilter<"StoryNode"> | $Enums.StoryNodeType
  content?: Prisma.JsonNullableFilter<"StoryNode">
  media?: Prisma.JsonNullableFilter<"StoryNode">
  createdAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  outgoing?: Prisma.StoryTransitionListRelationFilter
  incoming?: Prisma.StoryTransitionListRelationFilter
}

export type StoryNodeOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  key?: Prisma.SortOrder
  title?: Prisma.SortOrderInput | Prisma.SortOrder
  synopsis?: Prisma.SortOrderInput | Prisma.SortOrder
  type?: Prisma.SortOrder
  content?: Prisma.SortOrderInput | Prisma.SortOrder
  media?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  story?: Prisma.TwineStoryOrderByWithRelationInput
  outgoing?: Prisma.StoryTransitionOrderByRelationAggregateInput
  incoming?: Prisma.StoryTransitionOrderByRelationAggregateInput
}

export type StoryNodeWhereUniqueInput = Prisma.AtLeast<{
  id?: string
  storyId_key?: Prisma.StoryNodeStoryIdKeyCompoundUniqueInput
  AND?: Prisma.StoryNodeWhereInput | Prisma.StoryNodeWhereInput[]
  OR?: Prisma.StoryNodeWhereInput[]
  NOT?: Prisma.StoryNodeWhereInput | Prisma.StoryNodeWhereInput[]
  storyId?: Prisma.StringFilter<"StoryNode"> | string
  key?: Prisma.StringFilter<"StoryNode"> | string
  title?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  synopsis?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  type?: Prisma.EnumStoryNodeTypeFilter<"StoryNode"> | $Enums.StoryNodeType
  content?: Prisma.JsonNullableFilter<"StoryNode">
  media?: Prisma.JsonNullableFilter<"StoryNode">
  createdAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
  story?: Prisma.XOR<Prisma.TwineStoryScalarRelationFilter, Prisma.TwineStoryWhereInput>
  outgoing?: Prisma.StoryTransitionListRelationFilter
  incoming?: Prisma.StoryTransitionListRelationFilter
}, "id" | "storyId_key">

export type StoryNodeOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  key?: Prisma.SortOrder
  title?: Prisma.SortOrderInput | Prisma.SortOrder
  synopsis?: Prisma.SortOrderInput | Prisma.SortOrder
  type?: Prisma.SortOrder
  content?: Prisma.SortOrderInput | Prisma.SortOrder
  media?: Prisma.SortOrderInput | Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  _count?: Prisma.StoryNodeCountOrderByAggregateInput
  _max?: Prisma.StoryNodeMaxOrderByAggregateInput
  _min?: Prisma.StoryNodeMinOrderByAggregateInput
}

export type StoryNodeScalarWhereWithAggregatesInput = {
  AND?: Prisma.StoryNodeScalarWhereWithAggregatesInput | Prisma.StoryNodeScalarWhereWithAggregatesInput[]
  OR?: Prisma.StoryNodeScalarWhereWithAggregatesInput[]
  NOT?: Prisma.StoryNodeScalarWhereWithAggregatesInput | Prisma.StoryNodeScalarWhereWithAggregatesInput[]
  id?: Prisma.StringWithAggregatesFilter<"StoryNode"> | string
  storyId?: Prisma.StringWithAggregatesFilter<"StoryNode"> | string
  key?: Prisma.StringWithAggregatesFilter<"StoryNode"> | string
  title?: Prisma.StringNullableWithAggregatesFilter<"StoryNode"> | string | null
  synopsis?: Prisma.StringNullableWithAggregatesFilter<"StoryNode"> | string | null
  type?: Prisma.EnumStoryNodeTypeWithAggregatesFilter<"StoryNode"> | $Enums.StoryNodeType
  content?: Prisma.JsonNullableWithAggregatesFilter<"StoryNode">
  media?: Prisma.JsonNullableWithAggregatesFilter<"StoryNode">
  createdAt?: Prisma.DateTimeWithAggregatesFilter<"StoryNode"> | Date | string
  updatedAt?: Prisma.DateTimeWithAggregatesFilter<"StoryNode"> | Date | string
}

export type StoryNodeCreateInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutNodesInput
  outgoing?: Prisma.StoryTransitionCreateNestedManyWithoutFromNodeInput
  incoming?: Prisma.StoryTransitionCreateNestedManyWithoutToNodeInput
}

export type StoryNodeUncheckedCreateInput = {
  id?: string
  storyId: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  outgoing?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutFromNodeInput
  incoming?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutToNodeInput
}

export type StoryNodeUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutNodesNestedInput
  outgoing?: Prisma.StoryTransitionUpdateManyWithoutFromNodeNestedInput
  incoming?: Prisma.StoryTransitionUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeUncheckedUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  outgoing?: Prisma.StoryTransitionUncheckedUpdateManyWithoutFromNodeNestedInput
  incoming?: Prisma.StoryTransitionUncheckedUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeCreateManyInput = {
  id?: string
  storyId: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryNodeUpdateManyMutationInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryNodeUncheckedUpdateManyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}

export type StoryNodeListRelationFilter = {
  every?: Prisma.StoryNodeWhereInput
  some?: Prisma.StoryNodeWhereInput
  none?: Prisma.StoryNodeWhereInput
}

export type StoryNodeOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type StoryNodeStoryIdKeyCompoundUniqueInput = {
  storyId: string
  key: string
}

export type StoryNodeCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  key?: Prisma.SortOrder
  title?: Prisma.SortOrder
  synopsis?: Prisma.SortOrder
  type?: Prisma.SortOrder
  content?: Prisma.SortOrder
  media?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryNodeMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  key?: Prisma.SortOrder
  title?: Prisma.SortOrder
  synopsis?: Prisma.SortOrder
  type?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryNodeMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  storyId?: Prisma.SortOrder
  key?: Prisma.SortOrder
  title?: Prisma.SortOrder
  synopsis?: Prisma.SortOrder
  type?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
}

export type StoryNodeScalarRelationFilter = {
  is?: Prisma.StoryNodeWhereInput
  isNot?: Prisma.StoryNodeWhereInput
}

export type StoryNodeNullableScalarRelationFilter = {
  is?: Prisma.StoryNodeWhereInput | null
  isNot?: Prisma.StoryNodeWhereInput | null
}

export type StoryNodeCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput> | Prisma.StoryNodeCreateWithoutStoryInput[] | Prisma.StoryNodeUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutStoryInput | Prisma.StoryNodeCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryNodeCreateManyStoryInputEnvelope
  connect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
}

export type StoryNodeUncheckedCreateNestedManyWithoutStoryInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput> | Prisma.StoryNodeCreateWithoutStoryInput[] | Prisma.StoryNodeUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutStoryInput | Prisma.StoryNodeCreateOrConnectWithoutStoryInput[]
  createMany?: Prisma.StoryNodeCreateManyStoryInputEnvelope
  connect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
}

export type StoryNodeUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput> | Prisma.StoryNodeCreateWithoutStoryInput[] | Prisma.StoryNodeUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutStoryInput | Prisma.StoryNodeCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryNodeUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryNodeUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryNodeCreateManyStoryInputEnvelope
  set?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  disconnect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  delete?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  connect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  update?: Prisma.StoryNodeUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryNodeUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryNodeUpdateManyWithWhereWithoutStoryInput | Prisma.StoryNodeUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryNodeScalarWhereInput | Prisma.StoryNodeScalarWhereInput[]
}

export type StoryNodeUncheckedUpdateManyWithoutStoryNestedInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput> | Prisma.StoryNodeCreateWithoutStoryInput[] | Prisma.StoryNodeUncheckedCreateWithoutStoryInput[]
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutStoryInput | Prisma.StoryNodeCreateOrConnectWithoutStoryInput[]
  upsert?: Prisma.StoryNodeUpsertWithWhereUniqueWithoutStoryInput | Prisma.StoryNodeUpsertWithWhereUniqueWithoutStoryInput[]
  createMany?: Prisma.StoryNodeCreateManyStoryInputEnvelope
  set?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  disconnect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  delete?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  connect?: Prisma.StoryNodeWhereUniqueInput | Prisma.StoryNodeWhereUniqueInput[]
  update?: Prisma.StoryNodeUpdateWithWhereUniqueWithoutStoryInput | Prisma.StoryNodeUpdateWithWhereUniqueWithoutStoryInput[]
  updateMany?: Prisma.StoryNodeUpdateManyWithWhereWithoutStoryInput | Prisma.StoryNodeUpdateManyWithWhereWithoutStoryInput[]
  deleteMany?: Prisma.StoryNodeScalarWhereInput | Prisma.StoryNodeScalarWhereInput[]
}

export type EnumStoryNodeTypeFieldUpdateOperationsInput = {
  set?: $Enums.StoryNodeType
}

export type StoryNodeCreateNestedOneWithoutOutgoingInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutOutgoingInput, Prisma.StoryNodeUncheckedCreateWithoutOutgoingInput>
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutOutgoingInput
  connect?: Prisma.StoryNodeWhereUniqueInput
}

export type StoryNodeCreateNestedOneWithoutIncomingInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutIncomingInput, Prisma.StoryNodeUncheckedCreateWithoutIncomingInput>
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutIncomingInput
  connect?: Prisma.StoryNodeWhereUniqueInput
}

export type StoryNodeUpdateOneRequiredWithoutOutgoingNestedInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutOutgoingInput, Prisma.StoryNodeUncheckedCreateWithoutOutgoingInput>
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutOutgoingInput
  upsert?: Prisma.StoryNodeUpsertWithoutOutgoingInput
  connect?: Prisma.StoryNodeWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.StoryNodeUpdateToOneWithWhereWithoutOutgoingInput, Prisma.StoryNodeUpdateWithoutOutgoingInput>, Prisma.StoryNodeUncheckedUpdateWithoutOutgoingInput>
}

export type StoryNodeUpdateOneWithoutIncomingNestedInput = {
  create?: Prisma.XOR<Prisma.StoryNodeCreateWithoutIncomingInput, Prisma.StoryNodeUncheckedCreateWithoutIncomingInput>
  connectOrCreate?: Prisma.StoryNodeCreateOrConnectWithoutIncomingInput
  upsert?: Prisma.StoryNodeUpsertWithoutIncomingInput
  disconnect?: Prisma.StoryNodeWhereInput | boolean
  delete?: Prisma.StoryNodeWhereInput | boolean
  connect?: Prisma.StoryNodeWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.StoryNodeUpdateToOneWithWhereWithoutIncomingInput, Prisma.StoryNodeUpdateWithoutIncomingInput>, Prisma.StoryNodeUncheckedUpdateWithoutIncomingInput>
}

export type StoryNodeCreateWithoutStoryInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  outgoing?: Prisma.StoryTransitionCreateNestedManyWithoutFromNodeInput
  incoming?: Prisma.StoryTransitionCreateNestedManyWithoutToNodeInput
}

export type StoryNodeUncheckedCreateWithoutStoryInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  outgoing?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutFromNodeInput
  incoming?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutToNodeInput
}

export type StoryNodeCreateOrConnectWithoutStoryInput = {
  where: Prisma.StoryNodeWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput>
}

export type StoryNodeCreateManyStoryInputEnvelope = {
  data: Prisma.StoryNodeCreateManyStoryInput | Prisma.StoryNodeCreateManyStoryInput[]
  skipDuplicates?: boolean
}

export type StoryNodeUpsertWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryNodeWhereUniqueInput
  update: Prisma.XOR<Prisma.StoryNodeUpdateWithoutStoryInput, Prisma.StoryNodeUncheckedUpdateWithoutStoryInput>
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutStoryInput, Prisma.StoryNodeUncheckedCreateWithoutStoryInput>
}

export type StoryNodeUpdateWithWhereUniqueWithoutStoryInput = {
  where: Prisma.StoryNodeWhereUniqueInput
  data: Prisma.XOR<Prisma.StoryNodeUpdateWithoutStoryInput, Prisma.StoryNodeUncheckedUpdateWithoutStoryInput>
}

export type StoryNodeUpdateManyWithWhereWithoutStoryInput = {
  where: Prisma.StoryNodeScalarWhereInput
  data: Prisma.XOR<Prisma.StoryNodeUpdateManyMutationInput, Prisma.StoryNodeUncheckedUpdateManyWithoutStoryInput>
}

export type StoryNodeScalarWhereInput = {
  AND?: Prisma.StoryNodeScalarWhereInput | Prisma.StoryNodeScalarWhereInput[]
  OR?: Prisma.StoryNodeScalarWhereInput[]
  NOT?: Prisma.StoryNodeScalarWhereInput | Prisma.StoryNodeScalarWhereInput[]
  id?: Prisma.StringFilter<"StoryNode"> | string
  storyId?: Prisma.StringFilter<"StoryNode"> | string
  key?: Prisma.StringFilter<"StoryNode"> | string
  title?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  synopsis?: Prisma.StringNullableFilter<"StoryNode"> | string | null
  type?: Prisma.EnumStoryNodeTypeFilter<"StoryNode"> | $Enums.StoryNodeType
  content?: Prisma.JsonNullableFilter<"StoryNode">
  media?: Prisma.JsonNullableFilter<"StoryNode">
  createdAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"StoryNode"> | Date | string
}

export type StoryNodeCreateWithoutOutgoingInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutNodesInput
  incoming?: Prisma.StoryTransitionCreateNestedManyWithoutToNodeInput
}

export type StoryNodeUncheckedCreateWithoutOutgoingInput = {
  id?: string
  storyId: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  incoming?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutToNodeInput
}

export type StoryNodeCreateOrConnectWithoutOutgoingInput = {
  where: Prisma.StoryNodeWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutOutgoingInput, Prisma.StoryNodeUncheckedCreateWithoutOutgoingInput>
}

export type StoryNodeCreateWithoutIncomingInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  story: Prisma.TwineStoryCreateNestedOneWithoutNodesInput
  outgoing?: Prisma.StoryTransitionCreateNestedManyWithoutFromNodeInput
}

export type StoryNodeUncheckedCreateWithoutIncomingInput = {
  id?: string
  storyId: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
  outgoing?: Prisma.StoryTransitionUncheckedCreateNestedManyWithoutFromNodeInput
}

export type StoryNodeCreateOrConnectWithoutIncomingInput = {
  where: Prisma.StoryNodeWhereUniqueInput
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutIncomingInput, Prisma.StoryNodeUncheckedCreateWithoutIncomingInput>
}

export type StoryNodeUpsertWithoutOutgoingInput = {
  update: Prisma.XOR<Prisma.StoryNodeUpdateWithoutOutgoingInput, Prisma.StoryNodeUncheckedUpdateWithoutOutgoingInput>
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutOutgoingInput, Prisma.StoryNodeUncheckedCreateWithoutOutgoingInput>
  where?: Prisma.StoryNodeWhereInput
}

export type StoryNodeUpdateToOneWithWhereWithoutOutgoingInput = {
  where?: Prisma.StoryNodeWhereInput
  data: Prisma.XOR<Prisma.StoryNodeUpdateWithoutOutgoingInput, Prisma.StoryNodeUncheckedUpdateWithoutOutgoingInput>
}

export type StoryNodeUpdateWithoutOutgoingInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutNodesNestedInput
  incoming?: Prisma.StoryTransitionUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeUncheckedUpdateWithoutOutgoingInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  incoming?: Prisma.StoryTransitionUncheckedUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeUpsertWithoutIncomingInput = {
  update: Prisma.XOR<Prisma.StoryNodeUpdateWithoutIncomingInput, Prisma.StoryNodeUncheckedUpdateWithoutIncomingInput>
  create: Prisma.XOR<Prisma.StoryNodeCreateWithoutIncomingInput, Prisma.StoryNodeUncheckedCreateWithoutIncomingInput>
  where?: Prisma.StoryNodeWhereInput
}

export type StoryNodeUpdateToOneWithWhereWithoutIncomingInput = {
  where?: Prisma.StoryNodeWhereInput
  data: Prisma.XOR<Prisma.StoryNodeUpdateWithoutIncomingInput, Prisma.StoryNodeUncheckedUpdateWithoutIncomingInput>
}

export type StoryNodeUpdateWithoutIncomingInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  story?: Prisma.TwineStoryUpdateOneRequiredWithoutNodesNestedInput
  outgoing?: Prisma.StoryTransitionUpdateManyWithoutFromNodeNestedInput
}

export type StoryNodeUncheckedUpdateWithoutIncomingInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  storyId?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  outgoing?: Prisma.StoryTransitionUncheckedUpdateManyWithoutFromNodeNestedInput
}

export type StoryNodeCreateManyStoryInput = {
  id?: string
  key: string
  title?: string | null
  synopsis?: string | null
  type?: $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Date | string
  updatedAt?: Date | string
}

export type StoryNodeUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  outgoing?: Prisma.StoryTransitionUpdateManyWithoutFromNodeNestedInput
  incoming?: Prisma.StoryTransitionUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeUncheckedUpdateWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  outgoing?: Prisma.StoryTransitionUncheckedUpdateManyWithoutFromNodeNestedInput
  incoming?: Prisma.StoryTransitionUncheckedUpdateManyWithoutToNodeNestedInput
}

export type StoryNodeUncheckedUpdateManyWithoutStoryInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  key?: Prisma.StringFieldUpdateOperationsInput | string
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  synopsis?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  type?: Prisma.EnumStoryNodeTypeFieldUpdateOperationsInput | $Enums.StoryNodeType
  content?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  media?: Prisma.NullableJsonNullValueInput | runtime.InputJsonValue
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
}


/**
 * Count Type StoryNodeCountOutputType
 */

export type StoryNodeCountOutputType = {
  outgoing: number
  incoming: number
}

export type StoryNodeCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  outgoing?: boolean | StoryNodeCountOutputTypeCountOutgoingArgs
  incoming?: boolean | StoryNodeCountOutputTypeCountIncomingArgs
}

/**
 * StoryNodeCountOutputType without action
 */
export type StoryNodeCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNodeCountOutputType
   */
  select?: Prisma.StoryNodeCountOutputTypeSelect<ExtArgs> | null
}

/**
 * StoryNodeCountOutputType without action
 */
export type StoryNodeCountOutputTypeCountOutgoingArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryTransitionWhereInput
}

/**
 * StoryNodeCountOutputType without action
 */
export type StoryNodeCountOutputTypeCountIncomingArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.StoryTransitionWhereInput
}


export type StoryNodeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  key?: boolean
  title?: boolean
  synopsis?: boolean
  type?: boolean
  content?: boolean
  media?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  outgoing?: boolean | Prisma.StoryNode$outgoingArgs<ExtArgs>
  incoming?: boolean | Prisma.StoryNode$incomingArgs<ExtArgs>
  _count?: boolean | Prisma.StoryNodeCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyNode"]>

export type StoryNodeSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  key?: boolean
  title?: boolean
  synopsis?: boolean
  type?: boolean
  content?: boolean
  media?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyNode"]>

export type StoryNodeSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  storyId?: boolean
  key?: boolean
  title?: boolean
  synopsis?: boolean
  type?: boolean
  content?: boolean
  media?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
}, ExtArgs["result"]["storyNode"]>

export type StoryNodeSelectScalar = {
  id?: boolean
  storyId?: boolean
  key?: boolean
  title?: boolean
  synopsis?: boolean
  type?: boolean
  content?: boolean
  media?: boolean
  createdAt?: boolean
  updatedAt?: boolean
}

export type StoryNodeOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "storyId" | "key" | "title" | "synopsis" | "type" | "content" | "media" | "createdAt" | "updatedAt", ExtArgs["result"]["storyNode"]>
export type StoryNodeInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
  outgoing?: boolean | Prisma.StoryNode$outgoingArgs<ExtArgs>
  incoming?: boolean | Prisma.StoryNode$incomingArgs<ExtArgs>
  _count?: boolean | Prisma.StoryNodeCountOutputTypeDefaultArgs<ExtArgs>
}
export type StoryNodeIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
}
export type StoryNodeIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  story?: boolean | Prisma.TwineStoryDefaultArgs<ExtArgs>
}

export type $StoryNodePayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "StoryNode"
  objects: {
    story: Prisma.$TwineStoryPayload<ExtArgs>
    outgoing: Prisma.$StoryTransitionPayload<ExtArgs>[]
    incoming: Prisma.$StoryTransitionPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: string
    storyId: string
    key: string
    title: string | null
    synopsis: string | null
    type: $Enums.StoryNodeType
    content: runtime.JsonValue | null
    media: runtime.JsonValue | null
    createdAt: Date
    updatedAt: Date
  }, ExtArgs["result"]["storyNode"]>
  composites: {}
}

export type StoryNodeGetPayload<S extends boolean | null | undefined | StoryNodeDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$StoryNodePayload, S>

export type StoryNodeCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<StoryNodeFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: StoryNodeCountAggregateInputType | true
  }

export interface StoryNodeDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['StoryNode'], meta: { name: 'StoryNode' } }
  /**
   * Find zero or one StoryNode that matches the filter.
   * @param {StoryNodeFindUniqueArgs} args - Arguments to find a StoryNode
   * @example
   * // Get one StoryNode
   * const storyNode = await prisma.storyNode.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends StoryNodeFindUniqueArgs>(args: Prisma.SelectSubset<T, StoryNodeFindUniqueArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one StoryNode that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {StoryNodeFindUniqueOrThrowArgs} args - Arguments to find a StoryNode
   * @example
   * // Get one StoryNode
   * const storyNode = await prisma.storyNode.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends StoryNodeFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, StoryNodeFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryNode that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeFindFirstArgs} args - Arguments to find a StoryNode
   * @example
   * // Get one StoryNode
   * const storyNode = await prisma.storyNode.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends StoryNodeFindFirstArgs>(args?: Prisma.SelectSubset<T, StoryNodeFindFirstArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first StoryNode that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeFindFirstOrThrowArgs} args - Arguments to find a StoryNode
   * @example
   * // Get one StoryNode
   * const storyNode = await prisma.storyNode.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends StoryNodeFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, StoryNodeFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more StoryNodes that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all StoryNodes
   * const storyNodes = await prisma.storyNode.findMany()
   * 
   * // Get first 10 StoryNodes
   * const storyNodes = await prisma.storyNode.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const storyNodeWithIdOnly = await prisma.storyNode.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends StoryNodeFindManyArgs>(args?: Prisma.SelectSubset<T, StoryNodeFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a StoryNode.
   * @param {StoryNodeCreateArgs} args - Arguments to create a StoryNode.
   * @example
   * // Create one StoryNode
   * const StoryNode = await prisma.storyNode.create({
   *   data: {
   *     // ... data to create a StoryNode
   *   }
   * })
   * 
   */
  create<T extends StoryNodeCreateArgs>(args: Prisma.SelectSubset<T, StoryNodeCreateArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many StoryNodes.
   * @param {StoryNodeCreateManyArgs} args - Arguments to create many StoryNodes.
   * @example
   * // Create many StoryNodes
   * const storyNode = await prisma.storyNode.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends StoryNodeCreateManyArgs>(args?: Prisma.SelectSubset<T, StoryNodeCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many StoryNodes and returns the data saved in the database.
   * @param {StoryNodeCreateManyAndReturnArgs} args - Arguments to create many StoryNodes.
   * @example
   * // Create many StoryNodes
   * const storyNode = await prisma.storyNode.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many StoryNodes and only return the `id`
   * const storyNodeWithIdOnly = await prisma.storyNode.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends StoryNodeCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, StoryNodeCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a StoryNode.
   * @param {StoryNodeDeleteArgs} args - Arguments to delete one StoryNode.
   * @example
   * // Delete one StoryNode
   * const StoryNode = await prisma.storyNode.delete({
   *   where: {
   *     // ... filter to delete one StoryNode
   *   }
   * })
   * 
   */
  delete<T extends StoryNodeDeleteArgs>(args: Prisma.SelectSubset<T, StoryNodeDeleteArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one StoryNode.
   * @param {StoryNodeUpdateArgs} args - Arguments to update one StoryNode.
   * @example
   * // Update one StoryNode
   * const storyNode = await prisma.storyNode.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends StoryNodeUpdateArgs>(args: Prisma.SelectSubset<T, StoryNodeUpdateArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more StoryNodes.
   * @param {StoryNodeDeleteManyArgs} args - Arguments to filter StoryNodes to delete.
   * @example
   * // Delete a few StoryNodes
   * const { count } = await prisma.storyNode.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends StoryNodeDeleteManyArgs>(args?: Prisma.SelectSubset<T, StoryNodeDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryNodes.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many StoryNodes
   * const storyNode = await prisma.storyNode.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends StoryNodeUpdateManyArgs>(args: Prisma.SelectSubset<T, StoryNodeUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more StoryNodes and returns the data updated in the database.
   * @param {StoryNodeUpdateManyAndReturnArgs} args - Arguments to update many StoryNodes.
   * @example
   * // Update many StoryNodes
   * const storyNode = await prisma.storyNode.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more StoryNodes and only return the `id`
   * const storyNodeWithIdOnly = await prisma.storyNode.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends StoryNodeUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, StoryNodeUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one StoryNode.
   * @param {StoryNodeUpsertArgs} args - Arguments to update or create a StoryNode.
   * @example
   * // Update or create a StoryNode
   * const storyNode = await prisma.storyNode.upsert({
   *   create: {
   *     // ... data to create a StoryNode
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the StoryNode we want to update
   *   }
   * })
   */
  upsert<T extends StoryNodeUpsertArgs>(args: Prisma.SelectSubset<T, StoryNodeUpsertArgs<ExtArgs>>): Prisma.Prisma__StoryNodeClient<runtime.Types.Result.GetResult<Prisma.$StoryNodePayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of StoryNodes.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeCountArgs} args - Arguments to filter StoryNodes to count.
   * @example
   * // Count the number of StoryNodes
   * const count = await prisma.storyNode.count({
   *   where: {
   *     // ... the filter for the StoryNodes we want to count
   *   }
   * })
  **/
  count<T extends StoryNodeCountArgs>(
    args?: Prisma.Subset<T, StoryNodeCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], StoryNodeCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a StoryNode.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends StoryNodeAggregateArgs>(args: Prisma.Subset<T, StoryNodeAggregateArgs>): Prisma.PrismaPromise<GetStoryNodeAggregateType<T>>

  /**
   * Group by StoryNode.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {StoryNodeGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends StoryNodeGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: StoryNodeGroupByArgs['orderBy'] }
      : { orderBy?: StoryNodeGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, StoryNodeGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetStoryNodeGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the StoryNode model
 */
readonly fields: StoryNodeFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for StoryNode.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__StoryNodeClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  story<T extends Prisma.TwineStoryDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.TwineStoryDefaultArgs<ExtArgs>>): Prisma.Prisma__TwineStoryClient<runtime.Types.Result.GetResult<Prisma.$TwineStoryPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  outgoing<T extends Prisma.StoryNode$outgoingArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryNode$outgoingArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  incoming<T extends Prisma.StoryNode$incomingArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.StoryNode$incomingArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$StoryTransitionPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the StoryNode model
 */
export interface StoryNodeFieldRefs {
  readonly id: Prisma.FieldRef<"StoryNode", 'String'>
  readonly storyId: Prisma.FieldRef<"StoryNode", 'String'>
  readonly key: Prisma.FieldRef<"StoryNode", 'String'>
  readonly title: Prisma.FieldRef<"StoryNode", 'String'>
  readonly synopsis: Prisma.FieldRef<"StoryNode", 'String'>
  readonly type: Prisma.FieldRef<"StoryNode", 'StoryNodeType'>
  readonly content: Prisma.FieldRef<"StoryNode", 'Json'>
  readonly media: Prisma.FieldRef<"StoryNode", 'Json'>
  readonly createdAt: Prisma.FieldRef<"StoryNode", 'DateTime'>
  readonly updatedAt: Prisma.FieldRef<"StoryNode", 'DateTime'>
}
    

// Custom InputTypes
/**
 * StoryNode findUnique
 */
export type StoryNodeFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter, which StoryNode to fetch.
   */
  where: Prisma.StoryNodeWhereUniqueInput
}

/**
 * StoryNode findUniqueOrThrow
 */
export type StoryNodeFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter, which StoryNode to fetch.
   */
  where: Prisma.StoryNodeWhereUniqueInput
}

/**
 * StoryNode findFirst
 */
export type StoryNodeFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter, which StoryNode to fetch.
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryNodes to fetch.
   */
  orderBy?: Prisma.StoryNodeOrderByWithRelationInput | Prisma.StoryNodeOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryNodes.
   */
  cursor?: Prisma.StoryNodeWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryNodes from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryNodes.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryNodes.
   */
  distinct?: Prisma.StoryNodeScalarFieldEnum | Prisma.StoryNodeScalarFieldEnum[]
}

/**
 * StoryNode findFirstOrThrow
 */
export type StoryNodeFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter, which StoryNode to fetch.
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryNodes to fetch.
   */
  orderBy?: Prisma.StoryNodeOrderByWithRelationInput | Prisma.StoryNodeOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for StoryNodes.
   */
  cursor?: Prisma.StoryNodeWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryNodes from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryNodes.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of StoryNodes.
   */
  distinct?: Prisma.StoryNodeScalarFieldEnum | Prisma.StoryNodeScalarFieldEnum[]
}

/**
 * StoryNode findMany
 */
export type StoryNodeFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter, which StoryNodes to fetch.
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of StoryNodes to fetch.
   */
  orderBy?: Prisma.StoryNodeOrderByWithRelationInput | Prisma.StoryNodeOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing StoryNodes.
   */
  cursor?: Prisma.StoryNodeWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` StoryNodes from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` StoryNodes.
   */
  skip?: number
  distinct?: Prisma.StoryNodeScalarFieldEnum | Prisma.StoryNodeScalarFieldEnum[]
}

/**
 * StoryNode create
 */
export type StoryNodeCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * The data needed to create a StoryNode.
   */
  data: Prisma.XOR<Prisma.StoryNodeCreateInput, Prisma.StoryNodeUncheckedCreateInput>
}

/**
 * StoryNode createMany
 */
export type StoryNodeCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many StoryNodes.
   */
  data: Prisma.StoryNodeCreateManyInput | Prisma.StoryNodeCreateManyInput[]
  skipDuplicates?: boolean
}

/**
 * StoryNode createManyAndReturn
 */
export type StoryNodeCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * The data used to create many StoryNodes.
   */
  data: Prisma.StoryNodeCreateManyInput | Prisma.StoryNodeCreateManyInput[]
  skipDuplicates?: boolean
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * StoryNode update
 */
export type StoryNodeUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * The data needed to update a StoryNode.
   */
  data: Prisma.XOR<Prisma.StoryNodeUpdateInput, Prisma.StoryNodeUncheckedUpdateInput>
  /**
   * Choose, which StoryNode to update.
   */
  where: Prisma.StoryNodeWhereUniqueInput
}

/**
 * StoryNode updateMany
 */
export type StoryNodeUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update StoryNodes.
   */
  data: Prisma.XOR<Prisma.StoryNodeUpdateManyMutationInput, Prisma.StoryNodeUncheckedUpdateManyInput>
  /**
   * Filter which StoryNodes to update
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * Limit how many StoryNodes to update.
   */
  limit?: number
}

/**
 * StoryNode updateManyAndReturn
 */
export type StoryNodeUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * The data used to update StoryNodes.
   */
  data: Prisma.XOR<Prisma.StoryNodeUpdateManyMutationInput, Prisma.StoryNodeUncheckedUpdateManyInput>
  /**
   * Filter which StoryNodes to update
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * Limit how many StoryNodes to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * StoryNode upsert
 */
export type StoryNodeUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * The filter to search for the StoryNode to update in case it exists.
   */
  where: Prisma.StoryNodeWhereUniqueInput
  /**
   * In case the StoryNode found by the `where` argument doesn't exist, create a new StoryNode with this data.
   */
  create: Prisma.XOR<Prisma.StoryNodeCreateInput, Prisma.StoryNodeUncheckedCreateInput>
  /**
   * In case the StoryNode was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.StoryNodeUpdateInput, Prisma.StoryNodeUncheckedUpdateInput>
}

/**
 * StoryNode delete
 */
export type StoryNodeDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
  /**
   * Filter which StoryNode to delete.
   */
  where: Prisma.StoryNodeWhereUniqueInput
}

/**
 * StoryNode deleteMany
 */
export type StoryNodeDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which StoryNodes to delete
   */
  where?: Prisma.StoryNodeWhereInput
  /**
   * Limit how many StoryNodes to delete.
   */
  limit?: number
}

/**
 * StoryNode.outgoing
 */
export type StoryNode$outgoingArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  where?: Prisma.StoryTransitionWhereInput
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.StoryTransitionScalarFieldEnum | Prisma.StoryTransitionScalarFieldEnum[]
}

/**
 * StoryNode.incoming
 */
export type StoryNode$incomingArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryTransition
   */
  select?: Prisma.StoryTransitionSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryTransition
   */
  omit?: Prisma.StoryTransitionOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryTransitionInclude<ExtArgs> | null
  where?: Prisma.StoryTransitionWhereInput
  orderBy?: Prisma.StoryTransitionOrderByWithRelationInput | Prisma.StoryTransitionOrderByWithRelationInput[]
  cursor?: Prisma.StoryTransitionWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.StoryTransitionScalarFieldEnum | Prisma.StoryTransitionScalarFieldEnum[]
}

/**
 * StoryNode without action
 */
export type StoryNodeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the StoryNode
   */
  select?: Prisma.StoryNodeSelect<ExtArgs> | null
  /**
   * Omit specific fields from the StoryNode
   */
  omit?: Prisma.StoryNodeOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.StoryNodeInclude<ExtArgs> | null
}
